// @sec-ant/zxing-wasm/dist/reader/index.js
import { g as pi, r as hi, d as N, a as _i } from "../core-5eb9bd55.js";
import { Z as Ti, b as Ai, c as Ei, p as Fi, s as Si } from "../core-5eb9bd55.js";
var Ve = (() => {
  var Y = import.meta.url;
  return function(q = {}) {
    var f = q, J, le;
    f.ready = new Promise((e, r) => {
      J = e, le = r;
    });
    var Be = Object.assign({}, f), Ne = "./this.program", Le = typeof window == "object", ve = typeof importScripts == "function";
    typeof process == "object" && typeof process.versions == "object" && process.versions.node;
    var H = "";
    function Wr(e) {
      return f.locateFile ? f.locateFile(e, H) : H + e;
    }
    var Fe;
    (Le || ve) && (ve ? H = self.location.href : typeof document < "u" && document.currentScript && (H = document.currentScript.src), Y && (H = Y), H.indexOf("blob:") !== 0 ? H = H.substr(0, H.replace(/[?#].*/, "").lastIndexOf("/") + 1) : H = "", ve && (Fe = (e) => {
      var r = new XMLHttpRequest();
      return r.open("GET", e, !1), r.responseType = "arraybuffer", r.send(null), new Uint8Array(r.response);
    })), f.print || console.log.bind(console);
    var re = f.printErr || console.error.bind(console);
    Object.assign(f, Be), Be = null, f.arguments && f.arguments, f.thisProgram && (Ne = f.thisProgram), f.quit && f.quit;
    var te;
    f.wasmBinary && (te = f.wasmBinary), f.noExitRuntime, typeof WebAssembly != "object" && he("no native wasm support detected");
    var de, P, ze = !1, W, D, K, pe, F, A, Xe, Ze;
    function Ge() {
      var e = de.buffer;
      f.HEAP8 = W = new Int8Array(e), f.HEAP16 = K = new Int16Array(e), f.HEAP32 = F = new Int32Array(e), f.HEAPU8 = D = new Uint8Array(e), f.HEAPU16 = pe = new Uint16Array(e), f.HEAPU32 = A = new Uint32Array(e), f.HEAPF32 = Xe = new Float32Array(e), f.HEAPF64 = Ze = new Float64Array(e);
    }
    var qe, Je = [], Ke = [], Qe = [];
    function Ur() {
      if (f.preRun)
        for (typeof f.preRun == "function" && (f.preRun = [f.preRun]); f.preRun.length; )
          xr(f.preRun.shift());
      Se(Je);
    }
    function Mr() {
      Se(Ke);
    }
    function Ir() {
      if (f.postRun)
        for (typeof f.postRun == "function" && (f.postRun = [f.postRun]); f.postRun.length; )
          Vr(f.postRun.shift());
      Se(Qe);
    }
    function xr(e) {
      Je.unshift(e);
    }
    function Yr(e) {
      Ke.unshift(e);
    }
    function Vr(e) {
      Qe.unshift(e);
    }
    var L = 0, ne = null;
    function Br(e) {
      L++, f.monitorRunDependencies && f.monitorRunDependencies(L);
    }
    function Nr(e) {
      if (L--, f.monitorRunDependencies && f.monitorRunDependencies(L), L == 0 && ne) {
        var r = ne;
        ne = null, r();
      }
    }
    function he(e) {
      f.onAbort && f.onAbort(e), e = "Aborted(" + e + ")", re(e), ze = !0, e += ". Build with -sASSERTIONS for more info.";
      var r = new WebAssembly.RuntimeError(e);
      throw le(r), r;
    }
    var Lr = "data:application/octet-stream;base64,";
    function er(e) {
      return e.startsWith(Lr);
    }
    var z;
    f.locateFile ? (z = "zxing_reader.wasm", er(z) || (z = Wr(z))) : z = new URL("/reader/zxing_reader.wasm", self.location).href;
    function rr(e) {
      if (e == z && te)
        return new Uint8Array(te);
      if (Fe)
        return Fe(e);
      throw "both async and sync fetching of the wasm failed";
    }
    function zr(e) {
      return !te && (Le || ve) && typeof fetch == "function" ? fetch(e, { credentials: "same-origin" }).then((r) => {
        if (!r.ok)
          throw "failed to load wasm binary file at '" + e + "'";
        return r.arrayBuffer();
      }).catch(() => rr(e)) : Promise.resolve().then(() => rr(e));
    }
    function tr(e, r, t) {
      return zr(e).then((n) => WebAssembly.instantiate(n, r)).then((n) => n).then(t, (n) => {
        re("failed to asynchronously prepare wasm: " + n), he(n);
      });
    }
    function Xr(e, r, t, n) {
      return !e && typeof WebAssembly.instantiateStreaming == "function" && !er(r) && typeof fetch == "function" ? fetch(r, { credentials: "same-origin" }).then((i) => {
        var a = WebAssembly.instantiateStreaming(i, t);
        return a.then(n, function(o) {
          return re("wasm streaming compile failed: " + o), re("falling back to ArrayBuffer instantiation"), tr(r, t, n);
        });
      }) : tr(r, t, n);
    }
    function Zr() {
      var e = { a: Hn };
      function r(n, i) {
        var a = n.exports;
        return P = a, de = P.qa, Ge(), qe = P.ua, Yr(P.ra), Nr(), a;
      }
      Br();
      function t(n) {
        r(n.instance);
      }
      if (f.instantiateWasm)
        try {
          return f.instantiateWasm(e, r);
        } catch (n) {
          re("Module.instantiateWasm callback failed with error: " + n), le(n);
        }
      return Xr(te, z, e, t).catch(le), {};
    }
    var Se = (e) => {
      for (; e.length > 0; )
        e.shift()(f);
    }, _e = [], ge = 0;
    function Gr(e) {
      var r = new ye(e);
      return r.get_caught() || (r.set_caught(!0), ge--), r.set_rethrown(!1), _e.push(r), Rr(r.excPtr), r.get_exception_ptr();
    }
    var U = 0;
    function qr() {
      _(0, 0);
      var e = _e.pop();
      Sr(e.excPtr), U = 0;
    }
    function ye(e) {
      this.excPtr = e, this.ptr = e - 24, this.set_type = function(r) {
        A[this.ptr + 4 >> 2] = r;
      }, this.get_type = function() {
        return A[this.ptr + 4 >> 2];
      }, this.set_destructor = function(r) {
        A[this.ptr + 8 >> 2] = r;
      }, this.get_destructor = function() {
        return A[this.ptr + 8 >> 2];
      }, this.set_caught = function(r) {
        r = r ? 1 : 0, W[this.ptr + 12 >> 0] = r;
      }, this.get_caught = function() {
        return W[this.ptr + 12 >> 0] != 0;
      }, this.set_rethrown = function(r) {
        r = r ? 1 : 0, W[this.ptr + 13 >> 0] = r;
      }, this.get_rethrown = function() {
        return W[this.ptr + 13 >> 0] != 0;
      }, this.init = function(r, t) {
        this.set_adjusted_ptr(0), this.set_type(r), this.set_destructor(t);
      }, this.set_adjusted_ptr = function(r) {
        A[this.ptr + 16 >> 2] = r;
      }, this.get_adjusted_ptr = function() {
        return A[this.ptr + 16 >> 2];
      }, this.get_exception_ptr = function() {
        var r = Dr(this.get_type());
        if (r)
          return A[this.excPtr >> 2];
        var t = this.get_adjusted_ptr();
        return t !== 0 ? t : this.excPtr;
      };
    }
    function Jr(e) {
      throw U || (U = e), U;
    }
    var Re = (e) => {
      var r = U;
      if (!r)
        return ce(0), 0;
      var t = new ye(r);
      t.set_adjusted_ptr(r);
      var n = t.get_type();
      if (!n)
        return ce(0), r;
      for (var i in e) {
        var a = e[i];
        if (a === 0 || a === n)
          break;
        var o = t.ptr + 16;
        if (kr(a, n, o))
          return ce(a), r;
      }
      return ce(n), r;
    }, Kr = () => Re([]), Qr = (e) => Re([e]), et = (e, r) => Re([e, r]);
    function rt(e) {
      var r = new ye(e).get_exception_ptr();
      return r;
    }
    function tt() {
      var e = _e.pop();
      e || he("no exception to throw");
      var r = e.excPtr;
      throw e.get_rethrown() || (_e.push(e), e.set_rethrown(!0), e.set_caught(!1), ge++), U = r, U;
    }
    function nt(e, r, t) {
      var n = new ye(e);
      throw n.init(r, t), U = e, ge++, U;
    }
    function it() {
      return ge;
    }
    var me = {};
    function nr(e) {
      for (; e.length; ) {
        var r = e.pop(), t = e.pop();
        t(r);
      }
    }
    function ie(e) {
      return this.fromWireType(F[e >> 2]);
    }
    var Q = {}, X = {}, we = {}, ir = void 0;
    function be(e) {
      throw new ir(e);
    }
    function Z(e, r, t) {
      e.forEach(function(s) {
        we[s] = r;
      });
      function n(s) {
        var u = t(s);
        u.length !== e.length && be("Mismatched type converter count");
        for (var c = 0; c < e.length; ++c)
          M(e[c], u[c]);
      }
      var i = new Array(r.length), a = [], o = 0;
      r.forEach((s, u) => {
        X.hasOwnProperty(s) ? i[u] = X[s] : (a.push(s), Q.hasOwnProperty(s) || (Q[s] = []), Q[s].push(() => {
          i[u] = X[s], ++o, o === a.length && n(i);
        }));
      }), a.length === 0 && n(i);
    }
    var at = function(e) {
      var r = me[e];
      delete me[e];
      var t = r.rawConstructor, n = r.rawDestructor, i = r.fields, a = i.map((o) => o.getterReturnType).concat(i.map((o) => o.setterArgumentType));
      Z([e], a, (o) => {
        var s = {};
        return i.forEach((u, c) => {
          var v = u.fieldName, p = o[c], h = u.getter, m = u.getterContext, $ = o[c + i.length], S = u.setter, C = u.setterContext;
          s[v] = { read: (R) => p.fromWireType(h(m, R)), write: (R, l) => {
            var d = [];
            S(C, R, $.toWireType(d, l)), nr(d);
          } };
        }), [{ name: r.name, fromWireType: function(u) {
          var c = {};
          for (var v in s)
            c[v] = s[v].read(u);
          return n(u), c;
        }, toWireType: function(u, c) {
          for (var v in s)
            if (!(v in c))
              throw new TypeError(`Missing field: "${v}"`);
          var p = t();
          for (v in s)
            s[v].write(p, c[v]);
          return u !== null && u.push(n, p), p;
        }, argPackAdvance: 8, readValueFromPointer: ie, destructorFunction: n }];
      });
    };
    function ot(e, r, t, n, i) {
    }
    function ke(e) {
      switch (e) {
        case 1:
          return 0;
        case 2:
          return 1;
        case 4:
          return 2;
        case 8:
          return 3;
        default:
          throw new TypeError(`Unknown type size: ${e}`);
      }
    }
    function st() {
      for (var e = new Array(256), r = 0; r < 256; ++r)
        e[r] = String.fromCharCode(r);
      ar = e;
    }
    var ar = void 0;
    function k(e) {
      for (var r = "", t = e; D[t]; )
        r += ar[D[t++]];
      return r;
    }
    var ee = void 0;
    function w(e) {
      throw new ee(e);
    }
    function ut(e, r, t = {}) {
      var n = r.name;
      if (e || w(`type "${n}" must have a positive integer typeid pointer`), X.hasOwnProperty(e)) {
        if (t.ignoreDuplicateRegistrations)
          return;
        w(`Cannot register type '${n}' twice`);
      }
      if (X[e] = r, delete we[e], Q.hasOwnProperty(e)) {
        var i = Q[e];
        delete Q[e], i.forEach((a) => a());
      }
    }
    function M(e, r, t = {}) {
      if (!("argPackAdvance" in r))
        throw new TypeError("registerType registeredInstance requires argPackAdvance");
      return ut(e, r, t);
    }
    function ft(e, r, t, n, i) {
      var a = ke(t);
      r = k(r), M(e, { name: r, fromWireType: function(o) {
        return !!o;
      }, toWireType: function(o, s) {
        return s ? n : i;
      }, argPackAdvance: 8, readValueFromPointer: function(o) {
        var s;
        if (t === 1)
          s = W;
        else if (t === 2)
          s = K;
        else if (t === 4)
          s = F;
        else
          throw new TypeError("Unknown boolean type size: " + r);
        return this.fromWireType(s[o >> a]);
      }, destructorFunction: null });
    }
    function ct(e) {
      if (!(this instanceof V) || !(e instanceof V))
        return !1;
      for (var r = this.$$.ptrType.registeredClass, t = this.$$.ptr, n = e.$$.ptrType.registeredClass, i = e.$$.ptr; r.baseClass; )
        t = r.upcast(t), r = r.baseClass;
      for (; n.baseClass; )
        i = n.upcast(i), n = n.baseClass;
      return r === n && t === i;
    }
    function lt(e) {
      return { count: e.count, deleteScheduled: e.deleteScheduled, preservePointerOnDelete: e.preservePointerOnDelete, ptr: e.ptr, ptrType: e.ptrType, smartPtr: e.smartPtr, smartPtrType: e.smartPtrType };
    }
    function De(e) {
      function r(t) {
        return t.$$.ptrType.registeredClass.name;
      }
      w(r(e) + " instance already deleted");
    }
    var Oe = !1;
    function or(e) {
    }
    function vt(e) {
      e.smartPtr ? e.smartPtrType.rawDestructor(e.smartPtr) : e.ptrType.registeredClass.rawDestructor(e.ptr);
    }
    function sr(e) {
      e.count.value -= 1;
      var r = e.count.value === 0;
      r && vt(e);
    }
    function ur(e, r, t) {
      if (r === t)
        return e;
      if (t.baseClass === void 0)
        return null;
      var n = ur(e, r, t.baseClass);
      return n === null ? null : t.downcast(n);
    }
    var fr = {};
    function dt() {
      return Object.keys(se).length;
    }
    function pt() {
      var e = [];
      for (var r in se)
        se.hasOwnProperty(r) && e.push(se[r]);
      return e;
    }
    var ae = [];
    function je() {
      for (; ae.length; ) {
        var e = ae.pop();
        e.$$.deleteScheduled = !1, e.delete();
      }
    }
    var oe = void 0;
    function ht(e) {
      oe = e, ae.length && oe && oe(je);
    }
    function _t() {
      f.getInheritedInstanceCount = dt, f.getLiveInheritedInstances = pt, f.flushPendingDeletes = je, f.setDelayFunction = ht;
    }
    var se = {};
    function gt(e, r) {
      for (r === void 0 && w("ptr should not be undefined"); e.baseClass; )
        r = e.upcast(r), e = e.baseClass;
      return r;
    }
    function yt(e, r) {
      return r = gt(e, r), se[r];
    }
    function $e(e, r) {
      (!r.ptrType || !r.ptr) && be("makeClassHandle requires ptr and ptrType");
      var t = !!r.smartPtrType, n = !!r.smartPtr;
      return t !== n && be("Both smartPtrType and smartPtr must be specified"), r.count = { value: 1 }, ue(Object.create(e, { $$: { value: r } }));
    }
    function mt(e) {
      var r = this.getPointee(e);
      if (!r)
        return this.destructor(e), null;
      var t = yt(this.registeredClass, r);
      if (t !== void 0) {
        if (t.$$.count.value === 0)
          return t.$$.ptr = r, t.$$.smartPtr = e, t.clone();
        var n = t.clone();
        return this.destructor(e), n;
      }
      function i() {
        return this.isSmartPointer ? $e(this.registeredClass.instancePrototype, { ptrType: this.pointeeType, ptr: r, smartPtrType: this, smartPtr: e }) : $e(this.registeredClass.instancePrototype, { ptrType: this, ptr: e });
      }
      var a = this.registeredClass.getActualType(r), o = fr[a];
      if (!o)
        return i.call(this);
      var s;
      this.isConst ? s = o.constPointerType : s = o.pointerType;
      var u = ur(r, this.registeredClass, s.registeredClass);
      return u === null ? i.call(this) : this.isSmartPointer ? $e(s.registeredClass.instancePrototype, { ptrType: s, ptr: u, smartPtrType: this, smartPtr: e }) : $e(s.registeredClass.instancePrototype, { ptrType: s, ptr: u });
    }
    var ue = function(e) {
      return typeof FinalizationRegistry > "u" ? (ue = (r) => r, e) : (Oe = new FinalizationRegistry((r) => {
        sr(r.$$);
      }), ue = (r) => {
        var t = r.$$, n = !!t.smartPtr;
        if (n) {
          var i = { $$: t };
          Oe.register(r, i, r);
        }
        return r;
      }, or = (r) => Oe.unregister(r), ue(e));
    };
    function wt() {
      if (this.$$.ptr || De(this), this.$$.preservePointerOnDelete)
        return this.$$.count.value += 1, this;
      var e = ue(Object.create(Object.getPrototypeOf(this), { $$: { value: lt(this.$$) } }));
      return e.$$.count.value += 1, e.$$.deleteScheduled = !1, e;
    }
    function bt() {
      this.$$.ptr || De(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && w("Object already scheduled for deletion"), or(this), sr(this.$$), this.$$.preservePointerOnDelete || (this.$$.smartPtr = void 0, this.$$.ptr = void 0);
    }
    function $t() {
      return !this.$$.ptr;
    }
    function Ct() {
      return this.$$.ptr || De(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && w("Object already scheduled for deletion"), ae.push(this), ae.length === 1 && oe && oe(je), this.$$.deleteScheduled = !0, this;
    }
    function Pt() {
      V.prototype.isAliasOf = ct, V.prototype.clone = wt, V.prototype.delete = bt, V.prototype.isDeleted = $t, V.prototype.deleteLater = Ct;
    }
    function V() {
    }
    var Tt = 48, At = 57;
    function cr(e) {
      if (e === void 0)
        return "_unknown";
      e = e.replace(/[^a-zA-Z0-9_]/g, "$");
      var r = e.charCodeAt(0);
      return r >= Tt && r <= At ? `_${e}` : e;
    }
    function lr(e, r) {
      return e = cr(e), { [e]: function() {
        return r.apply(this, arguments);
      } }[e];
    }
    function vr(e, r, t) {
      if (e[r].overloadTable === void 0) {
        var n = e[r];
        e[r] = function() {
          return e[r].overloadTable.hasOwnProperty(arguments.length) || w(`Function '${t}' called with an invalid number of arguments (${arguments.length}) - expects one of (${e[r].overloadTable})!`), e[r].overloadTable[arguments.length].apply(this, arguments);
        }, e[r].overloadTable = [], e[r].overloadTable[n.argCount] = n;
      }
    }
    function dr(e, r, t) {
      f.hasOwnProperty(e) ? ((t === void 0 || f[e].overloadTable !== void 0 && f[e].overloadTable[t] !== void 0) && w(`Cannot register public name '${e}' twice`), vr(f, e, e), f.hasOwnProperty(t) && w(`Cannot register multiple overloads of a function with the same number of arguments (${t})!`), f[e].overloadTable[t] = r) : (f[e] = r, t !== void 0 && (f[e].numArguments = t));
    }
    function Et(e, r, t, n, i, a, o, s) {
      this.name = e, this.constructor = r, this.instancePrototype = t, this.rawDestructor = n, this.baseClass = i, this.getActualType = a, this.upcast = o, this.downcast = s, this.pureVirtualFunctions = [];
    }
    function He(e, r, t) {
      for (; r !== t; )
        r.upcast || w(`Expected null or instance of ${t.name}, got an instance of ${r.name}`), e = r.upcast(e), r = r.baseClass;
      return e;
    }
    function Ft(e, r) {
      if (r === null)
        return this.isReference && w(`null is not a valid ${this.name}`), 0;
      r.$$ || w(`Cannot pass "${Me(r)}" as a ${this.name}`), r.$$.ptr || w(`Cannot pass deleted object as a pointer of type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = He(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function St(e, r) {
      var t;
      if (r === null)
        return this.isReference && w(`null is not a valid ${this.name}`), this.isSmartPointer ? (t = this.rawConstructor(), e !== null && e.push(this.rawDestructor, t), t) : 0;
      r.$$ || w(`Cannot pass "${Me(r)}" as a ${this.name}`), r.$$.ptr || w(`Cannot pass deleted object as a pointer of type ${this.name}`), !this.isConst && r.$$.ptrType.isConst && w(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
      var n = r.$$.ptrType.registeredClass;
      if (t = He(r.$$.ptr, n, this.registeredClass), this.isSmartPointer)
        switch (r.$$.smartPtr === void 0 && w("Passing raw pointer to smart pointer is illegal"), this.sharingPolicy) {
          case 0:
            r.$$.smartPtrType === this ? t = r.$$.smartPtr : w(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
            break;
          case 1:
            t = r.$$.smartPtr;
            break;
          case 2:
            if (r.$$.smartPtrType === this)
              t = r.$$.smartPtr;
            else {
              var i = r.clone();
              t = this.rawShare(t, B.toHandle(function() {
                i.delete();
              })), e !== null && e.push(this.rawDestructor, t);
            }
            break;
          default:
            w("Unsupporting sharing policy");
        }
      return t;
    }
    function Rt(e, r) {
      if (r === null)
        return this.isReference && w(`null is not a valid ${this.name}`), 0;
      r.$$ || w(`Cannot pass "${Me(r)}" as a ${this.name}`), r.$$.ptr || w(`Cannot pass deleted object as a pointer of type ${this.name}`), r.$$.ptrType.isConst && w(`Cannot convert argument of type ${r.$$.ptrType.name} to parameter type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = He(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function kt(e) {
      return this.rawGetPointee && (e = this.rawGetPointee(e)), e;
    }
    function Dt(e) {
      this.rawDestructor && this.rawDestructor(e);
    }
    function Ot(e) {
      e !== null && e.delete();
    }
    function jt() {
      I.prototype.getPointee = kt, I.prototype.destructor = Dt, I.prototype.argPackAdvance = 8, I.prototype.readValueFromPointer = ie, I.prototype.deleteObject = Ot, I.prototype.fromWireType = mt;
    }
    function I(e, r, t, n, i, a, o, s, u, c, v) {
      this.name = e, this.registeredClass = r, this.isReference = t, this.isConst = n, this.isSmartPointer = i, this.pointeeType = a, this.sharingPolicy = o, this.rawGetPointee = s, this.rawConstructor = u, this.rawShare = c, this.rawDestructor = v, !i && r.baseClass === void 0 ? n ? (this.toWireType = Ft, this.destructorFunction = null) : (this.toWireType = Rt, this.destructorFunction = null) : this.toWireType = St;
    }
    function pr(e, r, t) {
      f.hasOwnProperty(e) || be("Replacing nonexistant public symbol"), f[e].overloadTable !== void 0 && t !== void 0 ? f[e].overloadTable[t] = r : (f[e] = r, f[e].argCount = t);
    }
    var Ht = (e, r, t) => {
      var n = f["dynCall_" + e];
      return t && t.length ? n.apply(null, [r].concat(t)) : n.call(null, r);
    }, Ce = [], b = (e) => {
      var r = Ce[e];
      return r || (e >= Ce.length && (Ce.length = e + 1), Ce[e] = r = qe.get(e)), r;
    }, Wt = (e, r, t) => {
      if (e.includes("j"))
        return Ht(e, r, t);
      var n = b(r).apply(null, t);
      return n;
    }, Ut = (e, r) => {
      var t = [];
      return function() {
        return t.length = 0, Object.assign(t, arguments), Wt(e, r, t);
      };
    };
    function j(e, r) {
      e = k(e);
      function t() {
        return e.includes("j") ? Ut(e, r) : b(r);
      }
      var n = t();
      return typeof n != "function" && w(`unknown function pointer with signature ${e}: ${r}`), n;
    }
    function Mt(e, r) {
      var t = lr(r, function(n) {
        this.name = r, this.message = n;
        var i = new Error(n).stack;
        i !== void 0 && (this.stack = this.toString() + `
` + i.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return t.prototype = Object.create(e.prototype), t.prototype.constructor = t, t.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, t;
    }
    var hr = void 0;
    function _r(e) {
      var r = Fr(e), t = k(r);
      return x(r), t;
    }
    function Pe(e, r) {
      var t = [], n = {};
      function i(a) {
        if (!n[a] && !X[a]) {
          if (we[a]) {
            we[a].forEach(i);
            return;
          }
          t.push(a), n[a] = !0;
        }
      }
      throw r.forEach(i), new hr(`${e}: ` + t.map(_r).join([", "]));
    }
    function It(e, r, t, n, i, a, o, s, u, c, v, p, h) {
      v = k(v), a = j(i, a), s && (s = j(o, s)), c && (c = j(u, c)), h = j(p, h);
      var m = cr(v);
      dr(m, function() {
        Pe(`Cannot construct ${v} due to unbound types`, [n]);
      }), Z([e, r, t], n ? [n] : [], function($) {
        $ = $[0];
        var S, C;
        n ? (S = $.registeredClass, C = S.instancePrototype) : C = V.prototype;
        var R = lr(m, function() {
          if (Object.getPrototypeOf(this) !== l)
            throw new ee("Use 'new' to construct " + v);
          if (d.constructor_body === void 0)
            throw new ee(v + " has no accessible constructor");
          var Ee = d.constructor_body[arguments.length];
          if (Ee === void 0)
            throw new ee(`Tried to invoke ctor of ${v} with invalid number of parameters (${arguments.length}) - expected (${Object.keys(d.constructor_body).toString()}) parameters instead!`);
          return Ee.apply(this, arguments);
        }), l = Object.create(C, { constructor: { value: R } });
        R.prototype = l;
        var d = new Et(v, R, l, h, S, a, s, c);
        d.baseClass && (d.baseClass.__derivedClasses === void 0 && (d.baseClass.__derivedClasses = []), d.baseClass.__derivedClasses.push(d));
        var T = new I(v, d, !0, !1, !1), E = new I(v + "*", d, !1, !1, !1), G = new I(v + " const*", d, !1, !0, !1);
        return fr[e] = { pointerType: E, constPointerType: G }, pr(m, R), [T, E, G];
      });
    }
    function We(e, r) {
      for (var t = [], n = 0; n < e; n++)
        t.push(A[r + n * 4 >> 2]);
      return t;
    }
    function Ue(e, r, t, n, i, a) {
      var o = r.length;
      o < 2 && w("argTypes array size mismatch! Must at least get return value and 'this' types!");
      for (var s = r[1] !== null && t !== null, u = !1, c = 1; c < r.length; ++c)
        if (r[c] !== null && r[c].destructorFunction === void 0) {
          u = !0;
          break;
        }
      var v = r[0].name !== "void", p = o - 2, h = new Array(p), m = [], $ = [];
      return function() {
        arguments.length !== p && w(`function ${e} called with ${arguments.length} arguments, expected ${p} args!`), $.length = 0;
        var S;
        m.length = s ? 2 : 1, m[0] = i, s && (S = r[1].toWireType($, this), m[1] = S);
        for (var C = 0; C < p; ++C)
          h[C] = r[C + 2].toWireType($, arguments[C]), m.push(h[C]);
        var R = n.apply(null, m);
        function l(d) {
          if (u)
            nr($);
          else
            for (var T = s ? 1 : 2; T < r.length; T++) {
              var E = T === 1 ? S : h[T - 2];
              r[T].destructorFunction !== null && r[T].destructorFunction(E);
            }
          if (v)
            return r[0].fromWireType(d);
        }
        return l(R);
      };
    }
    function xt(e, r, t, n, i, a) {
      var o = We(r, t);
      i = j(n, i), Z([], [e], function(s) {
        s = s[0];
        var u = `constructor ${s.name}`;
        if (s.registeredClass.constructor_body === void 0 && (s.registeredClass.constructor_body = []), s.registeredClass.constructor_body[r - 1] !== void 0)
          throw new ee(`Cannot register multiple constructors with identical number of parameters (${r - 1}) for class '${s.name}'! Overload resolution is currently only performed using the parameter count, not actual type info!`);
        return s.registeredClass.constructor_body[r - 1] = () => {
          Pe(`Cannot construct ${s.name} due to unbound types`, o);
        }, Z([], o, function(c) {
          return c.splice(1, 0, null), s.registeredClass.constructor_body[r - 1] = Ue(u, c, null, i, a), [];
        }), [];
      });
    }
    function Yt(e, r, t, n, i, a, o, s, u) {
      var c = We(t, n);
      r = k(r), a = j(i, a), Z([], [e], function(v) {
        v = v[0];
        var p = `${v.name}.${r}`;
        r.startsWith("@@") && (r = Symbol[r.substring(2)]), s && v.registeredClass.pureVirtualFunctions.push(r);
        function h() {
          Pe(`Cannot call ${p} due to unbound types`, c);
        }
        var m = v.registeredClass.instancePrototype, $ = m[r];
        return $ === void 0 || $.overloadTable === void 0 && $.className !== v.name && $.argCount === t - 2 ? (h.argCount = t - 2, h.className = v.name, m[r] = h) : (vr(m, r, p), m[r].overloadTable[t - 2] = h), Z([], c, function(S) {
          var C = Ue(p, S, v, a, o);
          return m[r].overloadTable === void 0 ? (C.argCount = t - 2, m[r] = C) : m[r].overloadTable[t - 2] = C, [];
        }), [];
      });
    }
    function Vt() {
      Object.assign(gr.prototype, { get(e) {
        return this.allocated[e];
      }, has(e) {
        return this.allocated[e] !== void 0;
      }, allocate(e) {
        var r = this.freelist.pop() || this.allocated.length;
        return this.allocated[r] = e, r;
      }, free(e) {
        this.allocated[e] = void 0, this.freelist.push(e);
      } });
    }
    function gr() {
      this.allocated = [void 0], this.freelist = [];
    }
    var O = new gr();
    function yr(e) {
      e >= O.reserved && --O.get(e).refcount === 0 && O.free(e);
    }
    function Bt() {
      for (var e = 0, r = O.reserved; r < O.allocated.length; ++r)
        O.allocated[r] !== void 0 && ++e;
      return e;
    }
    function Nt() {
      O.allocated.push({ value: void 0 }, { value: null }, { value: !0 }, { value: !1 }), O.reserved = O.allocated.length, f.count_emval_handles = Bt;
    }
    var B = { toValue: (e) => (e || w("Cannot use deleted val. handle = " + e), O.get(e).value), toHandle: (e) => {
      switch (e) {
        case void 0:
          return 1;
        case null:
          return 2;
        case !0:
          return 3;
        case !1:
          return 4;
        default:
          return O.allocate({ refcount: 1, value: e });
      }
    } };
    function Lt(e, r) {
      r = k(r), M(e, { name: r, fromWireType: function(t) {
        var n = B.toValue(t);
        return yr(t), n;
      }, toWireType: function(t, n) {
        return B.toHandle(n);
      }, argPackAdvance: 8, readValueFromPointer: ie, destructorFunction: null });
    }
    function Me(e) {
      if (e === null)
        return "null";
      var r = typeof e;
      return r === "object" || r === "array" || r === "function" ? e.toString() : "" + e;
    }
    function zt(e, r) {
      switch (r) {
        case 2:
          return function(t) {
            return this.fromWireType(Xe[t >> 2]);
          };
        case 3:
          return function(t) {
            return this.fromWireType(Ze[t >> 3]);
          };
        default:
          throw new TypeError("Unknown float type: " + e);
      }
    }
    function Xt(e, r, t) {
      var n = ke(t);
      r = k(r), M(e, { name: r, fromWireType: function(i) {
        return i;
      }, toWireType: function(i, a) {
        return a;
      }, argPackAdvance: 8, readValueFromPointer: zt(r, n), destructorFunction: null });
    }
    function Zt(e, r, t, n, i, a, o) {
      var s = We(r, t);
      e = k(e), i = j(n, i), dr(e, function() {
        Pe(`Cannot call ${e} due to unbound types`, s);
      }, r - 1), Z([], s, function(u) {
        var c = [u[0], null].concat(u.slice(1));
        return pr(e, Ue(e, c, null, i, a), r - 1), [];
      });
    }
    function Gt(e, r, t) {
      switch (r) {
        case 0:
          return t ? function(i) {
            return W[i];
          } : function(i) {
            return D[i];
          };
        case 1:
          return t ? function(i) {
            return K[i >> 1];
          } : function(i) {
            return pe[i >> 1];
          };
        case 2:
          return t ? function(i) {
            return F[i >> 2];
          } : function(i) {
            return A[i >> 2];
          };
        default:
          throw new TypeError("Unknown integer type: " + e);
      }
    }
    function qt(e, r, t, n, i) {
      r = k(r);
      var a = ke(t), o = (p) => p;
      if (n === 0) {
        var s = 32 - 8 * t;
        o = (p) => p << s >>> s;
      }
      var u = r.includes("unsigned"), c = (p, h) => {
      }, v;
      u ? v = function(p, h) {
        return c(h, this.name), h >>> 0;
      } : v = function(p, h) {
        return c(h, this.name), h;
      }, M(e, { name: r, fromWireType: o, toWireType: v, argPackAdvance: 8, readValueFromPointer: Gt(r, a, n !== 0), destructorFunction: null });
    }
    function Jt(e, r, t) {
      var n = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], i = n[r];
      function a(o) {
        o = o >> 2;
        var s = A, u = s[o], c = s[o + 1];
        return new i(s.buffer, c, u);
      }
      t = k(t), M(e, { name: t, fromWireType: a, argPackAdvance: 8, readValueFromPointer: a }, { ignoreDuplicateRegistrations: !0 });
    }
    var mr = (e, r, t, n) => {
      if (!(n > 0))
        return 0;
      for (var i = t, a = t + n - 1, o = 0; o < e.length; ++o) {
        var s = e.charCodeAt(o);
        if (s >= 55296 && s <= 57343) {
          var u = e.charCodeAt(++o);
          s = 65536 + ((s & 1023) << 10) | u & 1023;
        }
        if (s <= 127) {
          if (t >= a)
            break;
          r[t++] = s;
        } else if (s <= 2047) {
          if (t + 1 >= a)
            break;
          r[t++] = 192 | s >> 6, r[t++] = 128 | s & 63;
        } else if (s <= 65535) {
          if (t + 2 >= a)
            break;
          r[t++] = 224 | s >> 12, r[t++] = 128 | s >> 6 & 63, r[t++] = 128 | s & 63;
        } else {
          if (t + 3 >= a)
            break;
          r[t++] = 240 | s >> 18, r[t++] = 128 | s >> 12 & 63, r[t++] = 128 | s >> 6 & 63, r[t++] = 128 | s & 63;
        }
      }
      return r[t] = 0, t - i;
    }, Kt = (e, r, t) => mr(e, D, r, t), wr = (e) => {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n <= 127 ? r++ : n <= 2047 ? r += 2 : n >= 55296 && n <= 57343 ? (r += 4, ++t) : r += 3;
      }
      return r;
    }, br = typeof TextDecoder < "u" ? new TextDecoder("utf8") : void 0, Qt = (e, r, t) => {
      for (var n = r + t, i = r; e[i] && !(i >= n); )
        ++i;
      if (i - r > 16 && e.buffer && br)
        return br.decode(e.subarray(r, i));
      for (var a = ""; r < i; ) {
        var o = e[r++];
        if (!(o & 128)) {
          a += String.fromCharCode(o);
          continue;
        }
        var s = e[r++] & 63;
        if ((o & 224) == 192) {
          a += String.fromCharCode((o & 31) << 6 | s);
          continue;
        }
        var u = e[r++] & 63;
        if ((o & 240) == 224 ? o = (o & 15) << 12 | s << 6 | u : o = (o & 7) << 18 | s << 12 | u << 6 | e[r++] & 63, o < 65536)
          a += String.fromCharCode(o);
        else {
          var c = o - 65536;
          a += String.fromCharCode(55296 | c >> 10, 56320 | c & 1023);
        }
      }
      return a;
    }, Ie = (e, r) => e ? Qt(D, e, r) : "";
    function en(e, r) {
      r = k(r);
      var t = r === "std::string";
      M(e, { name: r, fromWireType: function(n) {
        var i = A[n >> 2], a = n + 4, o;
        if (t)
          for (var s = a, u = 0; u <= i; ++u) {
            var c = a + u;
            if (u == i || D[c] == 0) {
              var v = c - s, p = Ie(s, v);
              o === void 0 ? o = p : (o += String.fromCharCode(0), o += p), s = c + 1;
            }
          }
        else {
          for (var h = new Array(i), u = 0; u < i; ++u)
            h[u] = String.fromCharCode(D[a + u]);
          o = h.join("");
        }
        return x(n), o;
      }, toWireType: function(n, i) {
        i instanceof ArrayBuffer && (i = new Uint8Array(i));
        var a, o = typeof i == "string";
        o || i instanceof Uint8Array || i instanceof Uint8ClampedArray || i instanceof Int8Array || w("Cannot pass non-string to std::string"), t && o ? a = wr(i) : a = i.length;
        var s = Ye(4 + a + 1), u = s + 4;
        if (A[s >> 2] = a, t && o)
          Kt(i, u, a + 1);
        else if (o)
          for (var c = 0; c < a; ++c) {
            var v = i.charCodeAt(c);
            v > 255 && (x(u), w("String has UTF-16 code units that do not fit in 8 bits")), D[u + c] = v;
          }
        else
          for (var c = 0; c < a; ++c)
            D[u + c] = i[c];
        return n !== null && n.push(x, s), s;
      }, argPackAdvance: 8, readValueFromPointer: ie, destructorFunction: function(n) {
        x(n);
      } });
    }
    var $r = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0, rn = (e, r) => {
      for (var t = e, n = t >> 1, i = n + r / 2; !(n >= i) && pe[n]; )
        ++n;
      if (t = n << 1, t - e > 32 && $r)
        return $r.decode(D.subarray(e, t));
      for (var a = "", o = 0; !(o >= r / 2); ++o) {
        var s = K[e + o * 2 >> 1];
        if (s == 0)
          break;
        a += String.fromCharCode(s);
      }
      return a;
    }, tn = (e, r, t) => {
      if (t === void 0 && (t = 2147483647), t < 2)
        return 0;
      t -= 2;
      for (var n = r, i = t < e.length * 2 ? t / 2 : e.length, a = 0; a < i; ++a) {
        var o = e.charCodeAt(a);
        K[r >> 1] = o, r += 2;
      }
      return K[r >> 1] = 0, r - n;
    }, nn = (e) => e.length * 2, an = (e, r) => {
      for (var t = 0, n = ""; !(t >= r / 4); ) {
        var i = F[e + t * 4 >> 2];
        if (i == 0)
          break;
        if (++t, i >= 65536) {
          var a = i - 65536;
          n += String.fromCharCode(55296 | a >> 10, 56320 | a & 1023);
        } else
          n += String.fromCharCode(i);
      }
      return n;
    }, on = (e, r, t) => {
      if (t === void 0 && (t = 2147483647), t < 4)
        return 0;
      for (var n = r, i = n + t - 4, a = 0; a < e.length; ++a) {
        var o = e.charCodeAt(a);
        if (o >= 55296 && o <= 57343) {
          var s = e.charCodeAt(++a);
          o = 65536 + ((o & 1023) << 10) | s & 1023;
        }
        if (F[r >> 2] = o, r += 4, r + 4 > i)
          break;
      }
      return F[r >> 2] = 0, r - n;
    }, sn = (e) => {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n >= 55296 && n <= 57343 && ++t, r += 4;
      }
      return r;
    }, un = function(e, r, t) {
      t = k(t);
      var n, i, a, o, s;
      r === 2 ? (n = rn, i = tn, o = nn, a = () => pe, s = 1) : r === 4 && (n = an, i = on, o = sn, a = () => A, s = 2), M(e, { name: t, fromWireType: function(u) {
        for (var c = A[u >> 2], v = a(), p, h = u + 4, m = 0; m <= c; ++m) {
          var $ = u + 4 + m * r;
          if (m == c || v[$ >> s] == 0) {
            var S = $ - h, C = n(h, S);
            p === void 0 ? p = C : (p += String.fromCharCode(0), p += C), h = $ + r;
          }
        }
        return x(u), p;
      }, toWireType: function(u, c) {
        typeof c != "string" && w(`Cannot pass non-string to C++ string type ${t}`);
        var v = o(c), p = Ye(4 + v + r);
        return A[p >> 2] = v >> s, i(c, p + 4, v + r), u !== null && u.push(x, p), p;
      }, argPackAdvance: 8, readValueFromPointer: ie, destructorFunction: function(u) {
        x(u);
      } });
    };
    function fn(e, r, t, n, i, a) {
      me[e] = { name: k(r), rawConstructor: j(t, n), rawDestructor: j(i, a), fields: [] };
    }
    function cn(e, r, t, n, i, a, o, s, u, c) {
      me[e].fields.push({ fieldName: k(r), getterReturnType: t, getter: j(n, i), getterContext: a, setterArgumentType: o, setter: j(s, u), setterContext: c });
    }
    function ln(e, r) {
      r = k(r), M(e, { isVoid: !0, name: r, argPackAdvance: 0, fromWireType: function() {
      }, toWireType: function(t, n) {
      } });
    }
    var vn = {};
    function dn(e) {
      var r = vn[e];
      return r === void 0 ? k(e) : r;
    }
    function Cr() {
      if (typeof globalThis == "object")
        return globalThis;
      function e(r) {
        r.$$$embind_global$$$ = r;
        var t = typeof $$$embind_global$$$ == "object" && r.$$$embind_global$$$ == r;
        return t || delete r.$$$embind_global$$$, t;
      }
      if (typeof $$$embind_global$$$ == "object" || (typeof global == "object" && e(global) ? $$$embind_global$$$ = global : typeof self == "object" && e(self) && ($$$embind_global$$$ = self), typeof $$$embind_global$$$ == "object"))
        return $$$embind_global$$$;
      throw Error("unable to get global object.");
    }
    function pn(e) {
      return e === 0 ? B.toHandle(Cr()) : (e = dn(e), B.toHandle(Cr()[e]));
    }
    function hn(e) {
      e > 4 && (O.get(e).refcount += 1);
    }
    function Pr(e, r) {
      var t = X[e];
      return t === void 0 && w(r + " has unknown type " + _r(e)), t;
    }
    function _n(e) {
      var r = new Array(e + 1);
      return function(t, n, i) {
        r[0] = t;
        for (var a = 0; a < e; ++a) {
          var o = Pr(A[n + a * 4 >> 2], "parameter " + a);
          r[a + 1] = o.readValueFromPointer(i), i += o.argPackAdvance;
        }
        var s = new (t.bind.apply(t, r))();
        return B.toHandle(s);
      };
    }
    var Tr = {};
    function gn(e, r, t, n) {
      e = B.toValue(e);
      var i = Tr[r];
      return i || (i = _n(r), Tr[r] = i), i(e, t, n);
    }
    function yn(e, r) {
      e = Pr(e, "_emval_take_value");
      var t = e.readValueFromPointer(r);
      return B.toHandle(t);
    }
    var mn = () => {
      he("");
    }, wn = (e, r, t) => D.copyWithin(e, r, r + t), bn = () => 2147483648, $n = (e) => {
      var r = de.buffer, t = e - r.byteLength + 65535 >>> 16;
      try {
        return de.grow(t), Ge(), 1;
      } catch {
      }
    }, Cn = (e) => {
      var r = D.length;
      e >>>= 0;
      var t = bn();
      if (e > t)
        return !1;
      for (var n = (u, c) => u + (c - u % c) % c, i = 1; i <= 4; i *= 2) {
        var a = r * (1 + 0.2 / i);
        a = Math.min(a, e + 100663296);
        var o = Math.min(t, n(Math.max(e, a), 65536)), s = $n(o);
        if (s)
          return !0;
      }
      return !1;
    }, xe = {}, Pn = () => Ne || "./this.program", fe = () => {
      if (!fe.strings) {
        var e = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8", r = { USER: "web_user", LOGNAME: "web_user", PATH: "/", PWD: "/", HOME: "/home/web_user", LANG: e, _: Pn() };
        for (var t in xe)
          xe[t] === void 0 ? delete r[t] : r[t] = xe[t];
        var n = [];
        for (var t in r)
          n.push(`${t}=${r[t]}`);
        fe.strings = n;
      }
      return fe.strings;
    }, Tn = (e, r) => {
      for (var t = 0; t < e.length; ++t)
        W[r++ >> 0] = e.charCodeAt(t);
      W[r >> 0] = 0;
    }, An = (e, r) => {
      var t = 0;
      return fe().forEach(function(n, i) {
        var a = r + t;
        A[e + i * 4 >> 2] = a, Tn(n, a), t += n.length + 1;
      }), 0;
    }, En = (e, r) => {
      var t = fe();
      A[e >> 2] = t.length;
      var n = 0;
      return t.forEach(function(i) {
        n += i.length + 1;
      }), A[r >> 2] = n, 0;
    };
    function Fn(e) {
      return e;
    }
    var Te = (e) => e % 4 === 0 && (e % 100 !== 0 || e % 400 === 0), Sn = (e, r) => {
      for (var t = 0, n = 0; n <= r; t += e[n++])
        ;
      return t;
    }, Ar = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Er = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Rn = (e, r) => {
      for (var t = new Date(e.getTime()); r > 0; ) {
        var n = Te(t.getFullYear()), i = t.getMonth(), a = (n ? Ar : Er)[i];
        if (r > a - t.getDate())
          r -= a - t.getDate() + 1, t.setDate(1), i < 11 ? t.setMonth(i + 1) : (t.setMonth(0), t.setFullYear(t.getFullYear() + 1));
        else
          return t.setDate(t.getDate() + r), t;
      }
      return t;
    };
    function kn(e, r, t) {
      var n = t > 0 ? t : wr(e) + 1, i = new Array(n), a = mr(e, i, 0, i.length);
      return r && (i.length = a), i;
    }
    var Dn = (e, r) => {
      W.set(e, r);
    }, On = (e, r, t, n) => {
      var i = F[n + 40 >> 2], a = { tm_sec: F[n >> 2], tm_min: F[n + 4 >> 2], tm_hour: F[n + 8 >> 2], tm_mday: F[n + 12 >> 2], tm_mon: F[n + 16 >> 2], tm_year: F[n + 20 >> 2], tm_wday: F[n + 24 >> 2], tm_yday: F[n + 28 >> 2], tm_isdst: F[n + 32 >> 2], tm_gmtoff: F[n + 36 >> 2], tm_zone: i ? Ie(i) : "" }, o = Ie(t), s = { "%c": "%a %b %d %H:%M:%S %Y", "%D": "%m/%d/%y", "%F": "%Y-%m-%d", "%h": "%b", "%r": "%I:%M:%S %p", "%R": "%H:%M", "%T": "%H:%M:%S", "%x": "%m/%d/%y", "%X": "%H:%M:%S", "%Ec": "%c", "%EC": "%C", "%Ex": "%m/%d/%y", "%EX": "%H:%M:%S", "%Ey": "%y", "%EY": "%Y", "%Od": "%d", "%Oe": "%e", "%OH": "%H", "%OI": "%I", "%Om": "%m", "%OM": "%M", "%OS": "%S", "%Ou": "%u", "%OU": "%U", "%OV": "%V", "%Ow": "%w", "%OW": "%W", "%Oy": "%y" };
      for (var u in s)
        o = o.replace(new RegExp(u, "g"), s[u]);
      var c = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"], v = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
      function p(l, d, T) {
        for (var E = typeof l == "number" ? l.toString() : l || ""; E.length < d; )
          E = T[0] + E;
        return E;
      }
      function h(l, d) {
        return p(l, d, "0");
      }
      function m(l, d) {
        function T(G) {
          return G < 0 ? -1 : G > 0 ? 1 : 0;
        }
        var E;
        return (E = T(l.getFullYear() - d.getFullYear())) === 0 && (E = T(l.getMonth() - d.getMonth())) === 0 && (E = T(l.getDate() - d.getDate())), E;
      }
      function $(l) {
        switch (l.getDay()) {
          case 0:
            return new Date(l.getFullYear() - 1, 11, 29);
          case 1:
            return l;
          case 2:
            return new Date(l.getFullYear(), 0, 3);
          case 3:
            return new Date(l.getFullYear(), 0, 2);
          case 4:
            return new Date(l.getFullYear(), 0, 1);
          case 5:
            return new Date(l.getFullYear() - 1, 11, 31);
          case 6:
            return new Date(l.getFullYear() - 1, 11, 30);
        }
      }
      function S(l) {
        var d = Rn(new Date(l.tm_year + 1900, 0, 1), l.tm_yday), T = new Date(d.getFullYear(), 0, 4), E = new Date(d.getFullYear() + 1, 0, 4), G = $(T), Ee = $(E);
        return m(G, d) <= 0 ? m(Ee, d) <= 0 ? d.getFullYear() + 1 : d.getFullYear() : d.getFullYear() - 1;
      }
      var C = { "%a": (l) => c[l.tm_wday].substring(0, 3), "%A": (l) => c[l.tm_wday], "%b": (l) => v[l.tm_mon].substring(0, 3), "%B": (l) => v[l.tm_mon], "%C": (l) => {
        var d = l.tm_year + 1900;
        return h(d / 100 | 0, 2);
      }, "%d": (l) => h(l.tm_mday, 2), "%e": (l) => p(l.tm_mday, 2, " "), "%g": (l) => S(l).toString().substring(2), "%G": (l) => S(l), "%H": (l) => h(l.tm_hour, 2), "%I": (l) => {
        var d = l.tm_hour;
        return d == 0 ? d = 12 : d > 12 && (d -= 12), h(d, 2);
      }, "%j": (l) => h(l.tm_mday + Sn(Te(l.tm_year + 1900) ? Ar : Er, l.tm_mon - 1), 3), "%m": (l) => h(l.tm_mon + 1, 2), "%M": (l) => h(l.tm_min, 2), "%n": () => `
`, "%p": (l) => l.tm_hour >= 0 && l.tm_hour < 12 ? "AM" : "PM", "%S": (l) => h(l.tm_sec, 2), "%t": () => "	", "%u": (l) => l.tm_wday || 7, "%U": (l) => {
        var d = l.tm_yday + 7 - l.tm_wday;
        return h(Math.floor(d / 7), 2);
      }, "%V": (l) => {
        var d = Math.floor((l.tm_yday + 7 - (l.tm_wday + 6) % 7) / 7);
        if ((l.tm_wday + 371 - l.tm_yday - 2) % 7 <= 2 && d++, d) {
          if (d == 53) {
            var E = (l.tm_wday + 371 - l.tm_yday) % 7;
            E != 4 && (E != 3 || !Te(l.tm_year)) && (d = 1);
          }
        } else {
          d = 52;
          var T = (l.tm_wday + 7 - l.tm_yday - 1) % 7;
          (T == 4 || T == 5 && Te(l.tm_year % 400 - 1)) && d++;
        }
        return h(d, 2);
      }, "%w": (l) => l.tm_wday, "%W": (l) => {
        var d = l.tm_yday + 7 - (l.tm_wday + 6) % 7;
        return h(Math.floor(d / 7), 2);
      }, "%y": (l) => (l.tm_year + 1900).toString().substring(2), "%Y": (l) => l.tm_year + 1900, "%z": (l) => {
        var d = l.tm_gmtoff, T = d >= 0;
        return d = Math.abs(d) / 60, d = d / 60 * 100 + d % 60, (T ? "+" : "-") + ("0000" + d).slice(-4);
      }, "%Z": (l) => l.tm_zone, "%%": () => "%" };
      o = o.replace(/%%/g, "\0\0");
      for (var u in C)
        o.includes(u) && (o = o.replace(new RegExp(u, "g"), C[u](a)));
      o = o.replace(/\0\0/g, "%");
      var R = kn(o, !1);
      return R.length > r ? 0 : (Dn(R, e), R.length - 1);
    }, jn = (e, r, t, n, i) => On(e, r, t, n);
    ir = f.InternalError = class extends Error {
      constructor(r) {
        super(r), this.name = "InternalError";
      }
    }, st(), ee = f.BindingError = class extends Error {
      constructor(r) {
        super(r), this.name = "BindingError";
      }
    }, Pt(), _t(), jt(), hr = f.UnboundTypeError = Mt(Error, "UnboundTypeError"), Vt(), Nt();
    var Hn = { q: Gr, u: qr, a: Kr, h: Qr, l: et, I: rt, P: tt, n: nt, ba: it, d: Jr, oa: at, Y: ot, fa: ft, na: It, ma: xt, D: Yt, ea: Lt, W: Xt, J: Zt, w: qt, s: Jt, V: en, L: un, Q: fn, pa: cn, ga: ln, U: yr, la: pn, R: hn, ia: gn, ka: yn, K: mn, da: wn, ca: Cn, $: An, aa: En, H: ei, T: fi, B: ti, p: Jn, b: Wn, C: Qn, ha: ii, c: Yn, j: Bn, i: In, x: ri, O: Kn, v: Gn, G: oi, N: si, A: ni, F: ci, Z: vi, X: di, k: Vn, f: xn, e: Mn, g: Un, M: ui, m: Zn, o: Nn, S: Ln, t: Xn, ja: qn, y: ai, r: zn, E: li, z: Fn, _: jn };
    Zr();
    var x = f._free = (e) => (x = f._free = P.sa)(e), Ye = f._malloc = (e) => (Ye = f._malloc = P.ta)(e), Fr = (e) => (Fr = P.va)(e);
    f.__embind_initialize_bindings = () => (f.__embind_initialize_bindings = P.wa)();
    var _ = (e, r) => (_ = P.xa)(e, r), ce = (e) => (ce = P.ya)(e), g = () => (g = P.za)(), y = (e) => (y = P.Aa)(e), Sr = (e) => (Sr = P.Ba)(e), Rr = (e) => (Rr = P.Ca)(e), kr = (e, r, t) => (kr = P.Da)(e, r, t), Dr = (e) => (Dr = P.Ea)(e);
    f.dynCall_viijii = (e, r, t, n, i, a, o) => (f.dynCall_viijii = P.Fa)(e, r, t, n, i, a, o);
    var Or = f.dynCall_jiii = (e, r, t, n) => (Or = f.dynCall_jiii = P.Ga)(e, r, t, n), jr = f.dynCall_jiiii = (e, r, t, n, i) => (jr = f.dynCall_jiiii = P.Ha)(e, r, t, n, i);
    f.dynCall_iiiiij = (e, r, t, n, i, a, o) => (f.dynCall_iiiiij = P.Ia)(e, r, t, n, i, a, o), f.dynCall_iiiiijj = (e, r, t, n, i, a, o, s, u) => (f.dynCall_iiiiijj = P.Ja)(e, r, t, n, i, a, o, s, u), f.dynCall_iiiiiijj = (e, r, t, n, i, a, o, s, u, c) => (f.dynCall_iiiiiijj = P.Ka)(e, r, t, n, i, a, o, s, u, c);
    function Wn(e, r) {
      var t = g();
      try {
        return b(e)(r);
      } catch (n) {
        if (y(t), n !== n + 0)
          throw n;
        _(1, 0);
      }
    }
    function Un(e, r, t, n) {
      var i = g();
      try {
        b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function Mn(e, r, t) {
      var n = g();
      try {
        b(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function In(e, r, t, n, i) {
      var a = g();
      try {
        return b(e)(r, t, n, i);
      } catch (o) {
        if (y(a), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function xn(e, r) {
      var t = g();
      try {
        b(e)(r);
      } catch (n) {
        if (y(t), n !== n + 0)
          throw n;
        _(1, 0);
      }
    }
    function Yn(e, r, t) {
      var n = g();
      try {
        return b(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function Vn(e) {
      var r = g();
      try {
        b(e)();
      } catch (t) {
        if (y(r), t !== t + 0)
          throw t;
        _(1, 0);
      }
    }
    function Bn(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function Nn(e, r, t, n, i, a) {
      var o = g();
      try {
        b(e)(r, t, n, i, a);
      } catch (s) {
        if (y(o), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function Ln(e, r, t, n, i, a, o) {
      var s = g();
      try {
        b(e)(r, t, n, i, a, o);
      } catch (u) {
        if (y(s), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function zn(e, r, t, n, i, a, o, s, u, c, v) {
      var p = g();
      try {
        b(e)(r, t, n, i, a, o, s, u, c, v);
      } catch (h) {
        if (y(p), h !== h + 0)
          throw h;
        _(1, 0);
      }
    }
    function Xn(e, r, t, n, i, a, o, s) {
      var u = g();
      try {
        b(e)(r, t, n, i, a, o, s);
      } catch (c) {
        if (y(u), c !== c + 0)
          throw c;
        _(1, 0);
      }
    }
    function Zn(e, r, t, n, i) {
      var a = g();
      try {
        b(e)(r, t, n, i);
      } catch (o) {
        if (y(a), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function Gn(e, r, t, n, i, a, o) {
      var s = g();
      try {
        return b(e)(r, t, n, i, a, o);
      } catch (u) {
        if (y(s), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function qn(e, r, t, n, i, a, o, s, u) {
      var c = g();
      try {
        b(e)(r, t, n, i, a, o, s, u);
      } catch (v) {
        if (y(c), v !== v + 0)
          throw v;
        _(1, 0);
      }
    }
    function Jn(e) {
      var r = g();
      try {
        return b(e)();
      } catch (t) {
        if (y(r), t !== t + 0)
          throw t;
        _(1, 0);
      }
    }
    function Kn(e, r, t, n, i, a, o) {
      var s = g();
      try {
        return b(e)(r, t, n, i, a, o);
      } catch (u) {
        if (y(s), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function Qn(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function ei(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function ri(e, r, t, n, i, a) {
      var o = g();
      try {
        return b(e)(r, t, n, i, a);
      } catch (s) {
        if (y(o), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function ti(e, r, t, n, i, a) {
      var o = g();
      try {
        return b(e)(r, t, n, i, a);
      } catch (s) {
        if (y(o), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function ni(e, r, t, n, i, a, o, s, u, c) {
      var v = g();
      try {
        return b(e)(r, t, n, i, a, o, s, u, c);
      } catch (p) {
        if (y(v), p !== p + 0)
          throw p;
        _(1, 0);
      }
    }
    function ii(e, r, t) {
      var n = g();
      try {
        return b(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function ai(e, r, t, n, i, a, o, s, u, c) {
      var v = g();
      try {
        b(e)(r, t, n, i, a, o, s, u, c);
      } catch (p) {
        if (y(v), p !== p + 0)
          throw p;
        _(1, 0);
      }
    }
    function oi(e, r, t, n, i, a, o, s) {
      var u = g();
      try {
        return b(e)(r, t, n, i, a, o, s);
      } catch (c) {
        if (y(u), c !== c + 0)
          throw c;
        _(1, 0);
      }
    }
    function si(e, r, t, n, i, a, o, s, u) {
      var c = g();
      try {
        return b(e)(r, t, n, i, a, o, s, u);
      } catch (v) {
        if (y(c), v !== v + 0)
          throw v;
        _(1, 0);
      }
    }
    function ui(e, r, t, n, i, a, o) {
      var s = g();
      try {
        b(e)(r, t, n, i, a, o);
      } catch (u) {
        if (y(s), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function fi(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function ci(e, r, t, n, i, a, o, s, u, c, v, p) {
      var h = g();
      try {
        return b(e)(r, t, n, i, a, o, s, u, c, v, p);
      } catch (m) {
        if (y(h), m !== m + 0)
          throw m;
        _(1, 0);
      }
    }
    function li(e, r, t, n, i, a, o, s, u, c, v, p, h, m, $, S) {
      var C = g();
      try {
        b(e)(r, t, n, i, a, o, s, u, c, v, p, h, m, $, S);
      } catch (R) {
        if (y(C), R !== R + 0)
          throw R;
        _(1, 0);
      }
    }
    function vi(e, r, t, n) {
      var i = g();
      try {
        return Or(e, r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function di(e, r, t, n, i) {
      var a = g();
      try {
        return jr(e, r, t, n, i);
      } catch (o) {
        if (y(a), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    var Ae;
    ne = function e() {
      Ae || Hr(), Ae || (ne = e);
    };
    function Hr() {
      if (L > 0 || (Ur(), L > 0))
        return;
      function e() {
        Ae || (Ae = !0, f.calledRun = !0, !ze && (Mr(), J(f), f.onRuntimeInitialized && f.onRuntimeInitialized(), Ir()));
      }
      f.setStatus ? (f.setStatus("Running..."), setTimeout(function() {
        setTimeout(function() {
          f.setStatus("");
        }, 1), e();
      }, 1)) : e();
    }
    if (f.preInit)
      for (typeof f.preInit == "function" && (f.preInit = [f.preInit]); f.preInit.length > 0; )
        f.preInit.pop()();
    return Hr(), q.ready;
  };
})();
function wi(Y) {
  return pi(Ve, Y);
}
async function bi(Y, {
  tryHarder: q = N.tryHarder,
  formats: f = N.formats,
  maxSymbols: J = N.maxSymbols
} = N) {
  return hi(
    Y,
    {
      tryHarder: q,
      formats: f,
      maxSymbols: J
    },
    Ve
  );
}
async function $i(Y, {
  tryHarder: q = N.tryHarder,
  formats: f = N.formats,
  maxSymbols: J = N.maxSymbols
} = N) {
  return _i(
    Y,
    {
      tryHarder: q,
      formats: f,
      maxSymbols: J
    },
    Ve
  );
}
export {
  Ti as ZXING_BARCODE_FORMAT_NAMES,
  Ai as ZXING_CHARACTOR_SET_NAMES,
  N as defaultZXingReadOptions,
  Ei as defaultZXingWriteOptions,
  wi as getZXingModule,
  Fi as purgeZXingModule,
  $i as readBarcodesFromImageData,
  bi as readBarcodesFromImageFile,
  Si as setZXingModuleOverrides
};
