// @sec-ant/barcode-detector/dist/pure.js
var Ie = (f, p, u) => {
  if (!p.has(f))
    throw TypeError("Cannot " + u);
};
var He = (f, p, u) => (Ie(f, p, "read from private field"), u ? u.call(f) : p.get(f)), Ue = (f, p, u) => {
  if (p.has(f))
    throw TypeError("Cannot add the same private member more than once");
  p instanceof WeakSet ? p.add(f) : p.set(f, u);
}, ze = (f, p, u, C) => (Ie(f, p, "write to private field"), C ? C.call(f, u) : p.set(f, u), u);
const Ve = [
  "aztec",
  "code_128",
  "code_39",
  "code_93",
  "codabar",
  "data_matrix",
  "ean_13",
  "ean_8",
  "itf",
  "pdf417",
  "qr_code",
  "upc_a",
  "upc_e",
  "unknown"
];
function Da(f) {
  if (Ge(f))
    return {
      width: f.naturalWidth,
      height: f.naturalHeight
    };
  if (Oa(f))
    return {
      width: f.width.baseVal.value,
      height: f.height.baseVal.value
    };
  if (Xe(f))
    return {
      width: f.videoWidth,
      height: f.videoHeight
    };
  if (qe(f))
    return {
      width: f.width,
      height: f.height
    };
  if (Je(f))
    return {
      width: f.width,
      height: f.height
    };
  if (ja(f))
    return {
      width: f.width,
      height: f.height
    };
  if (Ra(f))
    return {
      width: f.displayWidth,
      height: f.displayHeight
    };
  throw new Error("Unsupported image type");
}
function Ge(f) {
  try {
    return f instanceof HTMLImageElement;
  } catch {
    return !1;
  }
}
function Oa(f) {
  try {
    return f instanceof SVGImageElement;
  } catch {
    return !1;
  }
}
function Xe(f) {
  try {
    return f instanceof HTMLVideoElement;
  } catch {
    return !1;
  }
}
function qe(f) {
  try {
    return f instanceof HTMLCanvasElement;
  } catch {
    return !1;
  }
}
function Je(f) {
  try {
    return f instanceof ImageBitmap;
  } catch {
    return !1;
  }
}
function ja(f) {
  try {
    return f instanceof OffscreenCanvas;
  } catch {
    return !1;
  }
}
function Ra(f) {
  try {
    return f instanceof VideoFrame;
  } catch {
    return !1;
  }
}
function Wa(f) {
  try {
    return f instanceof Blob;
  } catch {
    return !1;
  }
}
function ka(f) {
  try {
    return f instanceof ImageData;
  } catch {
    return !1;
  }
}
function Ia(f, p) {
  try {
    const u = new OffscreenCanvas(f, p);
    if (u.getContext("2d") instanceof OffscreenCanvasRenderingContext2D)
      return u;
    throw new Error("OffscreenCanvasRenderingContext2D is not supported");
  } catch {
    const u = document.createElement("canvas");
    return u.width = f, u.height = p, u;
  }
}
async function Qe(f) {
  if (Ge(f) && !await za(f))
    throw new DOMException("Failed to decode the image.", "InvalidStateError");
  if (Xe(f) && (f.readyState === 0 || f.readyState === 1))
    throw new DOMException(
      "The video data is not available.",
      "InvalidStateError"
    );
  if (Je(f) && Ba(f))
    throw new DOMException("The image bitmap is closed.", "InvalidStateError");
  const { width: p, height: u } = Da(f);
  if (p === 0 || u === 0)
    return null;
  const C = Ia(p, u), P = C.getContext("2d");
  P.drawImage(f, 0, 0);
  const M = P.getImageData(0, 0, p, u);
  return qe(C) && C.remove(), M;
}
async function Ha(f) {
  let p;
  try {
    p = await createImageBitmap(f);
  } catch (C) {
    throw C;
  }
  return await Qe(p);
}
async function Ua(f) {
  if (Wa(f))
    return await Ha(f);
  if (ka(f)) {
    if (Va(f.data.buffer))
      throw new DOMException(
        "The corresponding array buffer is detached.",
        "InvalidStateError"
      );
    return f;
  }
  return await Qe(f);
}
async function za(f) {
  try {
    return await f.decode(), !0;
  } catch {
    return !1;
  }
}
function Va(f) {
  return f.byteLength === 0;
}
function Ba(f) {
  return f.width === 0 && f.height === 0;
}
const Be = (f) => {
  let p;
  const u = /* @__PURE__ */ new Set(), C = (O, N) => {
    const H = typeof O == "function" ? O(p) : O;
    if (!Object.is(H, p)) {
      const j = p;
      p = N ?? typeof H != "object" ? H : Object.assign({}, p, H), u.forEach((Y) => Y(p, j));
    }
  }, P = () => p, M = { setState: C, getState: P, subscribe: (O) => (u.add(O), () => u.delete(O)), destroy: () => {
    u.clear();
  } };
  return p = f(C, P, M), M;
}, Ya = (f) => f ? Be(f) : Be, La = {
  locateFile: (f, p) => {
    var u;
    const C = (u = f.match(new RegExp("(?<=_).+?(?=\\.wasm$)"))) == null ? void 0 : u[0];
    return C ? `https://cdn.jsdelivr.net/npm/@sec-ant/zxing-wasm@2.1.2/dist/${C}/${f}` : p + f;
  }
}, lt = Ya()(() => ({
  zxingModuleWeakMap: /* @__PURE__ */ new WeakMap(),
  zxingModuleOverrides: La
}));
function Za(f) {
  lt.setState({
    zxingModuleOverrides: f
  });
}
function Ze(f, p = lt.getState().zxingModuleOverrides) {
  const { zxingModuleWeakMap: u } = lt.getState(), C = u.get(
    f
  );
  if (C && Object.is(p, lt.getState().zxingModuleOverrides))
    return C;
  {
    lt.setState({
      zxingModuleOverrides: p
    });
    const P = f(p);
    return u.set(f, P), P;
  }
}
const Ye = [
  "Aztec",
  "Codabar",
  "Code128",
  "Code39",
  "Code93",
  "DataBar",
  "DataBarExpanded",
  "DataMatrix",
  "EAN-13",
  "EAN-8",
  "ITF",
  "Linear-Codes",
  "Matrix-Codes",
  "MaxiCode",
  "MicroQRCode",
  "None",
  "PDF417",
  "QRCode",
  "UPC-A",
  "UPC-E"
], rt = {
  tryHarder: !0,
  formats: [],
  maxSymbols: 255
};
async function Na(f, {
  tryHarder: p = rt.tryHarder,
  formats: u = rt.formats,
  maxSymbols: C = rt.maxSymbols
} = rt, P) {
  const M = await Ze(
    P,
    lt.getState().zxingModuleOverrides
  ), {
    data: O,
    width: N,
    height: H,
    data: { byteLength: j }
  } = f, Y = M._malloc(j);
  M.HEAP8.set(O, Y);
  const G = M.readBarcodesFromPixmap(
    Y,
    N,
    H,
    p,
    Ga(u),
    C
  );
  M._free(Y);
  const z = [];
  for (let V = 0; V < G.size(); ++V) {
    const X = G.get(V);
    z.push({
      ...X,
      format: Xa(X.format)
    });
  }
  return z;
}
function Ga(f) {
  return f.join("|");
}
function Xa(f) {
  const p = Le(f);
  let u = 0, C = Ye.length - 1;
  for (; u <= C; ) {
    const P = Math.floor((u + C) / 2), M = Ye[P], O = Le(M);
    if (O === p)
      return M;
    O < p ? u = P + 1 : C = P - 1;
  }
  return "None";
}
function Le(f) {
  return f.toLowerCase().replace(/_-\[\]/g, "");
}
var Ke = (() => {
  var f = import.meta.url;
  return function(p = {}) {
    var u = p, C, P;
    u.ready = new Promise((t, e) => {
      C = t, P = e;
    });
    var M = Object.assign({}, u), O = "./this.program", N = typeof window == "object", H = typeof importScripts == "function";
    typeof process == "object" && typeof process.versions == "object" && process.versions.node;
    var j = "";
    function Y(t) {
      return u.locateFile ? u.locateFile(t, j) : j + t;
    }
    var G;
    (N || H) && (H ? j = self.location.href : typeof document < "u" && document.currentScript && (j = document.currentScript.src), f && (j = f), j.indexOf("blob:") !== 0 ? j = j.substr(0, j.replace(/[?#].*/, "").lastIndexOf("/") + 1) : j = "", H && (G = (t) => {
      var e = new XMLHttpRequest();
      return e.open("GET", t, !1), e.responseType = "arraybuffer", e.send(null), new Uint8Array(e.response);
    })), u.print || console.log.bind(console);
    var z = u.printErr || console.error.bind(console);
    Object.assign(u, M), M = null, u.arguments && u.arguments, u.thisProgram && (O = u.thisProgram), u.quit && u.quit;
    var V;
    u.wasmBinary && (V = u.wasmBinary), u.noExitRuntime, typeof WebAssembly != "object" && _t("no native wasm support detected");
    var X, A, bt = !1, q, U, ut, Ct, R, F, qt, Jt;
    function Qt() {
      var t = X.buffer;
      u.HEAP8 = q = new Int8Array(t), u.HEAP16 = ut = new Int16Array(t), u.HEAP32 = R = new Int32Array(t), u.HEAPU8 = U = new Uint8Array(t), u.HEAPU16 = Ct = new Uint16Array(t), u.HEAPU32 = F = new Uint32Array(t), u.HEAPF32 = qt = new Float32Array(t), u.HEAPF64 = Jt = new Float64Array(t);
    }
    var Zt, Kt = [], te = [], ee = [];
    function er() {
      if (u.preRun)
        for (typeof u.preRun == "function" && (u.preRun = [u.preRun]); u.preRun.length; )
          ar(u.preRun.shift());
      Wt(Kt);
    }
    function rr() {
      Wt(te);
    }
    function nr() {
      if (u.postRun)
        for (typeof u.postRun == "function" && (u.postRun = [u.postRun]); u.postRun.length; )
          ir(u.postRun.shift());
      Wt(ee);
    }
    function ar(t) {
      Kt.unshift(t);
    }
    function or(t) {
      te.unshift(t);
    }
    function ir(t) {
      ee.unshift(t);
    }
    var nt = 0, ht = null;
    function sr(t) {
      nt++, u.monitorRunDependencies && u.monitorRunDependencies(nt);
    }
    function ur(t) {
      if (nt--, u.monitorRunDependencies && u.monitorRunDependencies(nt), nt == 0 && ht) {
        var e = ht;
        ht = null, e();
      }
    }
    function _t(t) {
      u.onAbort && u.onAbort(t), t = "Aborted(" + t + ")", z(t), bt = !0, t += ". Build with -sASSERTIONS for more info.";
      var e = new WebAssembly.RuntimeError(t);
      throw P(e), e;
    }
    var cr = "data:application/octet-stream;base64,";
    function re(t) {
      return t.startsWith(cr);
    }
    var at;
    u.locateFile ? (at = "zxing_reader.wasm", re(at) || (at = Y(at))) : at = new URL("/reader/zxing_reader.wasm", self.location).href;
    function ne(t) {
      if (t == at && V)
        return new Uint8Array(V);
      if (G)
        return G(t);
      throw "both async and sync fetching of the wasm failed";
    }
    function fr(t) {
      return !V && (N || H) && typeof fetch == "function" ? fetch(t, { credentials: "same-origin" }).then((e) => {
        if (!e.ok)
          throw "failed to load wasm binary file at '" + t + "'";
        return e.arrayBuffer();
      }).catch(() => ne(t)) : Promise.resolve().then(() => ne(t));
    }
    function ae(t, e, r) {
      return fr(t).then((n) => WebAssembly.instantiate(n, e)).then((n) => n).then(r, (n) => {
        z("failed to asynchronously prepare wasm: " + n), _t(n);
      });
    }
    function lr(t, e, r, n) {
      return !t && typeof WebAssembly.instantiateStreaming == "function" && !re(e) && typeof fetch == "function" ? fetch(e, { credentials: "same-origin" }).then((a) => {
        var o = WebAssembly.instantiateStreaming(a, r);
        return o.then(n, function(i) {
          return z("wasm streaming compile failed: " + i), z("falling back to ArrayBuffer instantiation"), ae(e, r, n);
        });
      }) : ae(e, r, n);
    }
    function hr() {
      var t = { a: Kn };
      function e(n, a) {
        var o = n.exports;
        return A = o, X = A.qa, Qt(), Zt = A.ua, or(A.ra), ur(), o;
      }
      sr();
      function r(n) {
        e(n.instance);
      }
      if (u.instantiateWasm)
        try {
          return u.instantiateWasm(t, e);
        } catch (n) {
          z("Module.instantiateWasm callback failed with error: " + n), P(n);
        }
      return lr(V, at, t, r).catch(P), {};
    }
    var Wt = (t) => {
      for (; t.length > 0; )
        t.shift()(u);
    }, Tt = [], Pt = 0;
    function dr(t) {
      var e = new At(t);
      return e.get_caught() || (e.set_caught(!0), Pt--), e.set_rethrown(!1), Tt.push(e), De(e.excPtr), e.get_exception_ptr();
    }
    var J = 0;
    function pr() {
      g(0, 0);
      var t = Tt.pop();
      Me(t.excPtr), J = 0;
    }
    function At(t) {
      this.excPtr = t, this.ptr = t - 24, this.set_type = function(e) {
        F[this.ptr + 4 >> 2] = e;
      }, this.get_type = function() {
        return F[this.ptr + 4 >> 2];
      }, this.set_destructor = function(e) {
        F[this.ptr + 8 >> 2] = e;
      }, this.get_destructor = function() {
        return F[this.ptr + 8 >> 2];
      }, this.set_caught = function(e) {
        e = e ? 1 : 0, q[this.ptr + 12 >> 0] = e;
      }, this.get_caught = function() {
        return q[this.ptr + 12 >> 0] != 0;
      }, this.set_rethrown = function(e) {
        e = e ? 1 : 0, q[this.ptr + 13 >> 0] = e;
      }, this.get_rethrown = function() {
        return q[this.ptr + 13 >> 0] != 0;
      }, this.init = function(e, r) {
        this.set_adjusted_ptr(0), this.set_type(e), this.set_destructor(r);
      }, this.set_adjusted_ptr = function(e) {
        F[this.ptr + 16 >> 2] = e;
      }, this.get_adjusted_ptr = function() {
        return F[this.ptr + 16 >> 2];
      }, this.get_exception_ptr = function() {
        var e = je(this.get_type());
        if (e)
          return F[this.excPtr >> 2];
        var r = this.get_adjusted_ptr();
        return r !== 0 ? r : this.excPtr;
      };
    }
    function yr(t) {
      throw J || (J = t), J;
    }
    var kt = (t) => {
      var e = J;
      if (!e)
        return wt(0), 0;
      var r = new At(e);
      r.set_adjusted_ptr(e);
      var n = r.get_type();
      if (!n)
        return wt(0), e;
      for (var a in t) {
        var o = t[a];
        if (o === 0 || o === n)
          break;
        var i = r.ptr + 16;
        if (Oe(o, n, i))
          return wt(o), e;
      }
      return wt(n), e;
    }, vr = () => kt([]), mr = (t) => kt([t]), gr = (t, e) => kt([t, e]);
    function wr(t) {
      var e = new At(t).get_exception_ptr();
      return e;
    }
    function $r() {
      var t = Tt.pop();
      t || _t("no exception to throw");
      var e = t.excPtr;
      throw t.get_rethrown() || (Tt.push(t), t.set_rethrown(!0), t.set_caught(!1), Pt++), J = e, J;
    }
    function br(t, e, r) {
      var n = new At(t);
      throw n.init(e, r), J = t, Pt++, J;
    }
    function Cr() {
      return Pt;
    }
    var Et = {};
    function oe(t) {
      for (; t.length; ) {
        var e = t.pop(), r = t.pop();
        r(e);
      }
    }
    function dt(t) {
      return this.fromWireType(R[t >> 2]);
    }
    var ct = {}, ot = {}, xt = {}, ie = void 0;
    function St(t) {
      throw new ie(t);
    }
    function it(t, e, r) {
      t.forEach(function(s) {
        xt[s] = e;
      });
      function n(s) {
        var c = r(s);
        c.length !== t.length && St("Mismatched type converter count");
        for (var l = 0; l < t.length; ++l)
          Q(t[l], c[l]);
      }
      var a = new Array(e.length), o = [], i = 0;
      e.forEach((s, c) => {
        ot.hasOwnProperty(s) ? a[c] = ot[s] : (o.push(s), ct.hasOwnProperty(s) || (ct[s] = []), ct[s].push(() => {
          a[c] = ot[s], ++i, i === o.length && n(a);
        }));
      }), o.length === 0 && n(a);
    }
    var _r = function(t) {
      var e = Et[t];
      delete Et[t];
      var r = e.rawConstructor, n = e.rawDestructor, a = e.fields, o = a.map((i) => i.getterReturnType).concat(a.map((i) => i.setterArgumentType));
      it([t], o, (i) => {
        var s = {};
        return a.forEach((c, l) => {
          var d = c.fieldName, v = i[l], m = c.getter, b = c.getterContext, E = i[l + a.length], W = c.setter, x = c.setterContext;
          s[d] = { read: (k) => v.fromWireType(m(b, k)), write: (k, h) => {
            var y = [];
            W(x, k, E.toWireType(y, h)), oe(y);
          } };
        }), [{ name: e.name, fromWireType: function(c) {
          var l = {};
          for (var d in s)
            l[d] = s[d].read(c);
          return n(c), l;
        }, toWireType: function(c, l) {
          for (var d in s)
            if (!(d in l))
              throw new TypeError(`Missing field: "${d}"`);
          var v = r();
          for (d in s)
            s[d].write(v, l[d]);
          return c !== null && c.push(n, v), v;
        }, argPackAdvance: 8, readValueFromPointer: dt, destructorFunction: n }];
      });
    };
    function Tr(t, e, r, n, a) {
    }
    function It(t) {
      switch (t) {
        case 1:
          return 0;
        case 2:
          return 1;
        case 4:
          return 2;
        case 8:
          return 3;
        default:
          throw new TypeError(`Unknown type size: ${t}`);
      }
    }
    function Pr() {
      for (var t = new Array(256), e = 0; e < 256; ++e)
        t[e] = String.fromCharCode(e);
      se = t;
    }
    var se = void 0;
    function I(t) {
      for (var e = "", r = t; U[r]; )
        e += se[U[r++]];
      return e;
    }
    var ft = void 0;
    function _(t) {
      throw new ft(t);
    }
    function Ar(t, e, r = {}) {
      var n = e.name;
      if (t || _(`type "${n}" must have a positive integer typeid pointer`), ot.hasOwnProperty(t)) {
        if (r.ignoreDuplicateRegistrations)
          return;
        _(`Cannot register type '${n}' twice`);
      }
      if (ot[t] = e, delete xt[t], ct.hasOwnProperty(t)) {
        var a = ct[t];
        delete ct[t], a.forEach((o) => o());
      }
    }
    function Q(t, e, r = {}) {
      if (!("argPackAdvance" in e))
        throw new TypeError("registerType registeredInstance requires argPackAdvance");
      return Ar(t, e, r);
    }
    function Er(t, e, r, n, a) {
      var o = It(r);
      e = I(e), Q(t, { name: e, fromWireType: function(i) {
        return !!i;
      }, toWireType: function(i, s) {
        return s ? n : a;
      }, argPackAdvance: 8, readValueFromPointer: function(i) {
        var s;
        if (r === 1)
          s = q;
        else if (r === 2)
          s = ut;
        else if (r === 4)
          s = R;
        else
          throw new TypeError("Unknown boolean type size: " + e);
        return this.fromWireType(s[i >> o]);
      }, destructorFunction: null });
    }
    function xr(t) {
      if (!(this instanceof tt) || !(t instanceof tt))
        return !1;
      for (var e = this.$$.ptrType.registeredClass, r = this.$$.ptr, n = t.$$.ptrType.registeredClass, a = t.$$.ptr; e.baseClass; )
        r = e.upcast(r), e = e.baseClass;
      for (; n.baseClass; )
        a = n.upcast(a), n = n.baseClass;
      return e === n && r === a;
    }
    function Sr(t) {
      return { count: t.count, deleteScheduled: t.deleteScheduled, preservePointerOnDelete: t.preservePointerOnDelete, ptr: t.ptr, ptrType: t.ptrType, smartPtr: t.smartPtr, smartPtrType: t.smartPtrType };
    }
    function Ht(t) {
      function e(r) {
        return r.$$.ptrType.registeredClass.name;
      }
      _(e(t) + " instance already deleted");
    }
    var Ut = !1;
    function ue(t) {
    }
    function Fr(t) {
      t.smartPtr ? t.smartPtrType.rawDestructor(t.smartPtr) : t.ptrType.registeredClass.rawDestructor(t.ptr);
    }
    function ce(t) {
      t.count.value -= 1;
      var e = t.count.value === 0;
      e && Fr(t);
    }
    function fe(t, e, r) {
      if (e === r)
        return t;
      if (r.baseClass === void 0)
        return null;
      var n = fe(t, e, r.baseClass);
      return n === null ? null : r.downcast(n);
    }
    var le = {};
    function Mr() {
      return Object.keys(vt).length;
    }
    function Dr() {
      var t = [];
      for (var e in vt)
        vt.hasOwnProperty(e) && t.push(vt[e]);
      return t;
    }
    var pt = [];
    function zt() {
      for (; pt.length; ) {
        var t = pt.pop();
        t.$$.deleteScheduled = !1, t.delete();
      }
    }
    var yt = void 0;
    function Or(t) {
      yt = t, pt.length && yt && yt(zt);
    }
    function jr() {
      u.getInheritedInstanceCount = Mr, u.getLiveInheritedInstances = Dr, u.flushPendingDeletes = zt, u.setDelayFunction = Or;
    }
    var vt = {};
    function Rr(t, e) {
      for (e === void 0 && _("ptr should not be undefined"); t.baseClass; )
        e = t.upcast(e), t = t.baseClass;
      return e;
    }
    function Wr(t, e) {
      return e = Rr(t, e), vt[e];
    }
    function Ft(t, e) {
      (!e.ptrType || !e.ptr) && St("makeClassHandle requires ptr and ptrType");
      var r = !!e.smartPtrType, n = !!e.smartPtr;
      return r !== n && St("Both smartPtrType and smartPtr must be specified"), e.count = { value: 1 }, mt(Object.create(t, { $$: { value: e } }));
    }
    function kr(t) {
      var e = this.getPointee(t);
      if (!e)
        return this.destructor(t), null;
      var r = Wr(this.registeredClass, e);
      if (r !== void 0) {
        if (r.$$.count.value === 0)
          return r.$$.ptr = e, r.$$.smartPtr = t, r.clone();
        var n = r.clone();
        return this.destructor(t), n;
      }
      function a() {
        return this.isSmartPointer ? Ft(this.registeredClass.instancePrototype, { ptrType: this.pointeeType, ptr: e, smartPtrType: this, smartPtr: t }) : Ft(this.registeredClass.instancePrototype, { ptrType: this, ptr: t });
      }
      var o = this.registeredClass.getActualType(e), i = le[o];
      if (!i)
        return a.call(this);
      var s;
      this.isConst ? s = i.constPointerType : s = i.pointerType;
      var c = fe(e, this.registeredClass, s.registeredClass);
      return c === null ? a.call(this) : this.isSmartPointer ? Ft(s.registeredClass.instancePrototype, { ptrType: s, ptr: c, smartPtrType: this, smartPtr: t }) : Ft(s.registeredClass.instancePrototype, { ptrType: s, ptr: c });
    }
    var mt = function(t) {
      return typeof FinalizationRegistry > "u" ? (mt = (e) => e, t) : (Ut = new FinalizationRegistry((e) => {
        ce(e.$$);
      }), mt = (e) => {
        var r = e.$$, n = !!r.smartPtr;
        if (n) {
          var a = { $$: r };
          Ut.register(e, a, e);
        }
        return e;
      }, ue = (e) => Ut.unregister(e), mt(t));
    };
    function Ir() {
      if (this.$$.ptr || Ht(this), this.$$.preservePointerOnDelete)
        return this.$$.count.value += 1, this;
      var t = mt(Object.create(Object.getPrototypeOf(this), { $$: { value: Sr(this.$$) } }));
      return t.$$.count.value += 1, t.$$.deleteScheduled = !1, t;
    }
    function Hr() {
      this.$$.ptr || Ht(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && _("Object already scheduled for deletion"), ue(this), ce(this.$$), this.$$.preservePointerOnDelete || (this.$$.smartPtr = void 0, this.$$.ptr = void 0);
    }
    function Ur() {
      return !this.$$.ptr;
    }
    function zr() {
      return this.$$.ptr || Ht(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && _("Object already scheduled for deletion"), pt.push(this), pt.length === 1 && yt && yt(zt), this.$$.deleteScheduled = !0, this;
    }
    function Vr() {
      tt.prototype.isAliasOf = xr, tt.prototype.clone = Ir, tt.prototype.delete = Hr, tt.prototype.isDeleted = Ur, tt.prototype.deleteLater = zr;
    }
    function tt() {
    }
    var Br = 48, Yr = 57;
    function he(t) {
      if (t === void 0)
        return "_unknown";
      t = t.replace(/[^a-zA-Z0-9_]/g, "$");
      var e = t.charCodeAt(0);
      return e >= Br && e <= Yr ? `_${t}` : t;
    }
    function de(t, e) {
      return t = he(t), { [t]: function() {
        return e.apply(this, arguments);
      } }[t];
    }
    function pe(t, e, r) {
      if (t[e].overloadTable === void 0) {
        var n = t[e];
        t[e] = function() {
          return t[e].overloadTable.hasOwnProperty(arguments.length) || _(`Function '${r}' called with an invalid number of arguments (${arguments.length}) - expects one of (${t[e].overloadTable})!`), t[e].overloadTable[arguments.length].apply(this, arguments);
        }, t[e].overloadTable = [], t[e].overloadTable[n.argCount] = n;
      }
    }
    function ye(t, e, r) {
      u.hasOwnProperty(t) ? ((r === void 0 || u[t].overloadTable !== void 0 && u[t].overloadTable[r] !== void 0) && _(`Cannot register public name '${t}' twice`), pe(u, t, t), u.hasOwnProperty(r) && _(`Cannot register multiple overloads of a function with the same number of arguments (${r})!`), u[t].overloadTable[r] = e) : (u[t] = e, r !== void 0 && (u[t].numArguments = r));
    }
    function Lr(t, e, r, n, a, o, i, s) {
      this.name = t, this.constructor = e, this.instancePrototype = r, this.rawDestructor = n, this.baseClass = a, this.getActualType = o, this.upcast = i, this.downcast = s, this.pureVirtualFunctions = [];
    }
    function Vt(t, e, r) {
      for (; e !== r; )
        e.upcast || _(`Expected null or instance of ${r.name}, got an instance of ${e.name}`), t = e.upcast(t), e = e.baseClass;
      return t;
    }
    function Nr(t, e) {
      if (e === null)
        return this.isReference && _(`null is not a valid ${this.name}`), 0;
      e.$$ || _(`Cannot pass "${Lt(e)}" as a ${this.name}`), e.$$.ptr || _(`Cannot pass deleted object as a pointer of type ${this.name}`);
      var r = e.$$.ptrType.registeredClass, n = Vt(e.$$.ptr, r, this.registeredClass);
      return n;
    }
    function Gr(t, e) {
      var r;
      if (e === null)
        return this.isReference && _(`null is not a valid ${this.name}`), this.isSmartPointer ? (r = this.rawConstructor(), t !== null && t.push(this.rawDestructor, r), r) : 0;
      e.$$ || _(`Cannot pass "${Lt(e)}" as a ${this.name}`), e.$$.ptr || _(`Cannot pass deleted object as a pointer of type ${this.name}`), !this.isConst && e.$$.ptrType.isConst && _(`Cannot convert argument of type ${e.$$.smartPtrType ? e.$$.smartPtrType.name : e.$$.ptrType.name} to parameter type ${this.name}`);
      var n = e.$$.ptrType.registeredClass;
      if (r = Vt(e.$$.ptr, n, this.registeredClass), this.isSmartPointer)
        switch (e.$$.smartPtr === void 0 && _("Passing raw pointer to smart pointer is illegal"), this.sharingPolicy) {
          case 0:
            e.$$.smartPtrType === this ? r = e.$$.smartPtr : _(`Cannot convert argument of type ${e.$$.smartPtrType ? e.$$.smartPtrType.name : e.$$.ptrType.name} to parameter type ${this.name}`);
            break;
          case 1:
            r = e.$$.smartPtr;
            break;
          case 2:
            if (e.$$.smartPtrType === this)
              r = e.$$.smartPtr;
            else {
              var a = e.clone();
              r = this.rawShare(r, et.toHandle(function() {
                a.delete();
              })), t !== null && t.push(this.rawDestructor, r);
            }
            break;
          default:
            _("Unsupporting sharing policy");
        }
      return r;
    }
    function Xr(t, e) {
      if (e === null)
        return this.isReference && _(`null is not a valid ${this.name}`), 0;
      e.$$ || _(`Cannot pass "${Lt(e)}" as a ${this.name}`), e.$$.ptr || _(`Cannot pass deleted object as a pointer of type ${this.name}`), e.$$.ptrType.isConst && _(`Cannot convert argument of type ${e.$$.ptrType.name} to parameter type ${this.name}`);
      var r = e.$$.ptrType.registeredClass, n = Vt(e.$$.ptr, r, this.registeredClass);
      return n;
    }
    function qr(t) {
      return this.rawGetPointee && (t = this.rawGetPointee(t)), t;
    }
    function Jr(t) {
      this.rawDestructor && this.rawDestructor(t);
    }
    function Qr(t) {
      t !== null && t.delete();
    }
    function Zr() {
      Z.prototype.getPointee = qr, Z.prototype.destructor = Jr, Z.prototype.argPackAdvance = 8, Z.prototype.readValueFromPointer = dt, Z.prototype.deleteObject = Qr, Z.prototype.fromWireType = kr;
    }
    function Z(t, e, r, n, a, o, i, s, c, l, d) {
      this.name = t, this.registeredClass = e, this.isReference = r, this.isConst = n, this.isSmartPointer = a, this.pointeeType = o, this.sharingPolicy = i, this.rawGetPointee = s, this.rawConstructor = c, this.rawShare = l, this.rawDestructor = d, !a && e.baseClass === void 0 ? n ? (this.toWireType = Nr, this.destructorFunction = null) : (this.toWireType = Xr, this.destructorFunction = null) : this.toWireType = Gr;
    }
    function ve(t, e, r) {
      u.hasOwnProperty(t) || St("Replacing nonexistant public symbol"), u[t].overloadTable !== void 0 && r !== void 0 ? u[t].overloadTable[r] = e : (u[t] = e, u[t].argCount = r);
    }
    var Kr = (t, e, r) => {
      var n = u["dynCall_" + t];
      return r && r.length ? n.apply(null, [e].concat(r)) : n.call(null, e);
    }, Mt = [], T = (t) => {
      var e = Mt[t];
      return e || (t >= Mt.length && (Mt.length = t + 1), Mt[t] = e = Zt.get(t)), e;
    }, tn = (t, e, r) => {
      if (t.includes("j"))
        return Kr(t, e, r);
      var n = T(e).apply(null, r);
      return n;
    }, en = (t, e) => {
      var r = [];
      return function() {
        return r.length = 0, Object.assign(r, arguments), tn(t, e, r);
      };
    };
    function L(t, e) {
      t = I(t);
      function r() {
        return t.includes("j") ? en(t, e) : T(e);
      }
      var n = r();
      return typeof n != "function" && _(`unknown function pointer with signature ${t}: ${e}`), n;
    }
    function rn(t, e) {
      var r = de(e, function(n) {
        this.name = e, this.message = n;
        var a = new Error(n).stack;
        a !== void 0 && (this.stack = this.toString() + `
` + a.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return r.prototype = Object.create(t.prototype), r.prototype.constructor = r, r.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, r;
    }
    var me = void 0;
    function ge(t) {
      var e = Fe(t), r = I(e);
      return K(e), r;
    }
    function Dt(t, e) {
      var r = [], n = {};
      function a(o) {
        if (!n[o] && !ot[o]) {
          if (xt[o]) {
            xt[o].forEach(a);
            return;
          }
          r.push(o), n[o] = !0;
        }
      }
      throw e.forEach(a), new me(`${t}: ` + r.map(ge).join([", "]));
    }
    function nn(t, e, r, n, a, o, i, s, c, l, d, v, m) {
      d = I(d), o = L(a, o), s && (s = L(i, s)), l && (l = L(c, l)), m = L(v, m);
      var b = he(d);
      ye(b, function() {
        Dt(`Cannot construct ${d} due to unbound types`, [n]);
      }), it([t, e, r], n ? [n] : [], function(E) {
        E = E[0];
        var W, x;
        n ? (W = E.registeredClass, x = W.instancePrototype) : x = tt.prototype;
        var k = de(b, function() {
          if (Object.getPrototypeOf(this) !== h)
            throw new ft("Use 'new' to construct " + d);
          if (y.constructor_body === void 0)
            throw new ft(d + " has no accessible constructor");
          var Rt = y.constructor_body[arguments.length];
          if (Rt === void 0)
            throw new ft(`Tried to invoke ctor of ${d} with invalid number of parameters (${arguments.length}) - expected (${Object.keys(y.constructor_body).toString()}) parameters instead!`);
          return Rt.apply(this, arguments);
        }), h = Object.create(x, { constructor: { value: k } });
        k.prototype = h;
        var y = new Lr(d, k, h, m, W, o, s, l);
        y.baseClass && (y.baseClass.__derivedClasses === void 0 && (y.baseClass.__derivedClasses = []), y.baseClass.__derivedClasses.push(y));
        var S = new Z(d, y, !0, !1, !1), D = new Z(d + "*", y, !1, !1, !1), st = new Z(d + " const*", y, !1, !0, !1);
        return le[t] = { pointerType: D, constPointerType: st }, ve(b, k), [S, D, st];
      });
    }
    function Bt(t, e) {
      for (var r = [], n = 0; n < t; n++)
        r.push(F[e + n * 4 >> 2]);
      return r;
    }
    function Yt(t, e, r, n, a, o) {
      var i = e.length;
      i < 2 && _("argTypes array size mismatch! Must at least get return value and 'this' types!");
      for (var s = e[1] !== null && r !== null, c = !1, l = 1; l < e.length; ++l)
        if (e[l] !== null && e[l].destructorFunction === void 0) {
          c = !0;
          break;
        }
      var d = e[0].name !== "void", v = i - 2, m = new Array(v), b = [], E = [];
      return function() {
        arguments.length !== v && _(`function ${t} called with ${arguments.length} arguments, expected ${v} args!`), E.length = 0;
        var W;
        b.length = s ? 2 : 1, b[0] = a, s && (W = e[1].toWireType(E, this), b[1] = W);
        for (var x = 0; x < v; ++x)
          m[x] = e[x + 2].toWireType(E, arguments[x]), b.push(m[x]);
        var k = n.apply(null, b);
        function h(y) {
          if (c)
            oe(E);
          else
            for (var S = s ? 1 : 2; S < e.length; S++) {
              var D = S === 1 ? W : m[S - 2];
              e[S].destructorFunction !== null && e[S].destructorFunction(D);
            }
          if (d)
            return e[0].fromWireType(y);
        }
        return h(k);
      };
    }
    function an(t, e, r, n, a, o) {
      var i = Bt(e, r);
      a = L(n, a), it([], [t], function(s) {
        s = s[0];
        var c = `constructor ${s.name}`;
        if (s.registeredClass.constructor_body === void 0 && (s.registeredClass.constructor_body = []), s.registeredClass.constructor_body[e - 1] !== void 0)
          throw new ft(`Cannot register multiple constructors with identical number of parameters (${e - 1}) for class '${s.name}'! Overload resolution is currently only performed using the parameter count, not actual type info!`);
        return s.registeredClass.constructor_body[e - 1] = () => {
          Dt(`Cannot construct ${s.name} due to unbound types`, i);
        }, it([], i, function(l) {
          return l.splice(1, 0, null), s.registeredClass.constructor_body[e - 1] = Yt(c, l, null, a, o), [];
        }), [];
      });
    }
    function on(t, e, r, n, a, o, i, s, c) {
      var l = Bt(r, n);
      e = I(e), o = L(a, o), it([], [t], function(d) {
        d = d[0];
        var v = `${d.name}.${e}`;
        e.startsWith("@@") && (e = Symbol[e.substring(2)]), s && d.registeredClass.pureVirtualFunctions.push(e);
        function m() {
          Dt(`Cannot call ${v} due to unbound types`, l);
        }
        var b = d.registeredClass.instancePrototype, E = b[e];
        return E === void 0 || E.overloadTable === void 0 && E.className !== d.name && E.argCount === r - 2 ? (m.argCount = r - 2, m.className = d.name, b[e] = m) : (pe(b, e, v), b[e].overloadTable[r - 2] = m), it([], l, function(W) {
          var x = Yt(v, W, d, o, i);
          return b[e].overloadTable === void 0 ? (x.argCount = r - 2, b[e] = x) : b[e].overloadTable[r - 2] = x, [];
        }), [];
      });
    }
    function sn() {
      Object.assign(we.prototype, { get(t) {
        return this.allocated[t];
      }, has(t) {
        return this.allocated[t] !== void 0;
      }, allocate(t) {
        var e = this.freelist.pop() || this.allocated.length;
        return this.allocated[e] = t, e;
      }, free(t) {
        this.allocated[t] = void 0, this.freelist.push(t);
      } });
    }
    function we() {
      this.allocated = [void 0], this.freelist = [];
    }
    var B = new we();
    function $e(t) {
      t >= B.reserved && --B.get(t).refcount === 0 && B.free(t);
    }
    function un() {
      for (var t = 0, e = B.reserved; e < B.allocated.length; ++e)
        B.allocated[e] !== void 0 && ++t;
      return t;
    }
    function cn() {
      B.allocated.push({ value: void 0 }, { value: null }, { value: !0 }, { value: !1 }), B.reserved = B.allocated.length, u.count_emval_handles = un;
    }
    var et = { toValue: (t) => (t || _("Cannot use deleted val. handle = " + t), B.get(t).value), toHandle: (t) => {
      switch (t) {
        case void 0:
          return 1;
        case null:
          return 2;
        case !0:
          return 3;
        case !1:
          return 4;
        default:
          return B.allocate({ refcount: 1, value: t });
      }
    } };
    function fn(t, e) {
      e = I(e), Q(t, { name: e, fromWireType: function(r) {
        var n = et.toValue(r);
        return $e(r), n;
      }, toWireType: function(r, n) {
        return et.toHandle(n);
      }, argPackAdvance: 8, readValueFromPointer: dt, destructorFunction: null });
    }
    function Lt(t) {
      if (t === null)
        return "null";
      var e = typeof t;
      return e === "object" || e === "array" || e === "function" ? t.toString() : "" + t;
    }
    function ln(t, e) {
      switch (e) {
        case 2:
          return function(r) {
            return this.fromWireType(qt[r >> 2]);
          };
        case 3:
          return function(r) {
            return this.fromWireType(Jt[r >> 3]);
          };
        default:
          throw new TypeError("Unknown float type: " + t);
      }
    }
    function hn(t, e, r) {
      var n = It(r);
      e = I(e), Q(t, { name: e, fromWireType: function(a) {
        return a;
      }, toWireType: function(a, o) {
        return o;
      }, argPackAdvance: 8, readValueFromPointer: ln(e, n), destructorFunction: null });
    }
    function dn(t, e, r, n, a, o, i) {
      var s = Bt(e, r);
      t = I(t), a = L(n, a), ye(t, function() {
        Dt(`Cannot call ${t} due to unbound types`, s);
      }, e - 1), it([], s, function(c) {
        var l = [c[0], null].concat(c.slice(1));
        return ve(t, Yt(t, l, null, a, o), e - 1), [];
      });
    }
    function pn(t, e, r) {
      switch (e) {
        case 0:
          return r ? function(n) {
            return q[n];
          } : function(n) {
            return U[n];
          };
        case 1:
          return r ? function(n) {
            return ut[n >> 1];
          } : function(n) {
            return Ct[n >> 1];
          };
        case 2:
          return r ? function(n) {
            return R[n >> 2];
          } : function(n) {
            return F[n >> 2];
          };
        default:
          throw new TypeError("Unknown integer type: " + t);
      }
    }
    function yn(t, e, r, n, a) {
      e = I(e);
      var o = It(r), i = (v) => v;
      if (n === 0) {
        var s = 32 - 8 * r;
        i = (v) => v << s >>> s;
      }
      var c = e.includes("unsigned"), l = (v, m) => {
      }, d;
      c ? d = function(v, m) {
        return l(m, this.name), m >>> 0;
      } : d = function(v, m) {
        return l(m, this.name), m;
      }, Q(t, { name: e, fromWireType: i, toWireType: d, argPackAdvance: 8, readValueFromPointer: pn(e, o, n !== 0), destructorFunction: null });
    }
    function vn(t, e, r) {
      var n = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], a = n[e];
      function o(i) {
        i = i >> 2;
        var s = F, c = s[i], l = s[i + 1];
        return new a(s.buffer, l, c);
      }
      r = I(r), Q(t, { name: r, fromWireType: o, argPackAdvance: 8, readValueFromPointer: o }, { ignoreDuplicateRegistrations: !0 });
    }
    var be = (t, e, r, n) => {
      if (!(n > 0))
        return 0;
      for (var a = r, o = r + n - 1, i = 0; i < t.length; ++i) {
        var s = t.charCodeAt(i);
        if (s >= 55296 && s <= 57343) {
          var c = t.charCodeAt(++i);
          s = 65536 + ((s & 1023) << 10) | c & 1023;
        }
        if (s <= 127) {
          if (r >= o)
            break;
          e[r++] = s;
        } else if (s <= 2047) {
          if (r + 1 >= o)
            break;
          e[r++] = 192 | s >> 6, e[r++] = 128 | s & 63;
        } else if (s <= 65535) {
          if (r + 2 >= o)
            break;
          e[r++] = 224 | s >> 12, e[r++] = 128 | s >> 6 & 63, e[r++] = 128 | s & 63;
        } else {
          if (r + 3 >= o)
            break;
          e[r++] = 240 | s >> 18, e[r++] = 128 | s >> 12 & 63, e[r++] = 128 | s >> 6 & 63, e[r++] = 128 | s & 63;
        }
      }
      return e[r] = 0, r - a;
    }, mn = (t, e, r) => be(t, U, e, r), Ce = (t) => {
      for (var e = 0, r = 0; r < t.length; ++r) {
        var n = t.charCodeAt(r);
        n <= 127 ? e++ : n <= 2047 ? e += 2 : n >= 55296 && n <= 57343 ? (e += 4, ++r) : e += 3;
      }
      return e;
    }, _e = typeof TextDecoder < "u" ? new TextDecoder("utf8") : void 0, gn = (t, e, r) => {
      for (var n = e + r, a = e; t[a] && !(a >= n); )
        ++a;
      if (a - e > 16 && t.buffer && _e)
        return _e.decode(t.subarray(e, a));
      for (var o = ""; e < a; ) {
        var i = t[e++];
        if (!(i & 128)) {
          o += String.fromCharCode(i);
          continue;
        }
        var s = t[e++] & 63;
        if ((i & 224) == 192) {
          o += String.fromCharCode((i & 31) << 6 | s);
          continue;
        }
        var c = t[e++] & 63;
        if ((i & 240) == 224 ? i = (i & 15) << 12 | s << 6 | c : i = (i & 7) << 18 | s << 12 | c << 6 | t[e++] & 63, i < 65536)
          o += String.fromCharCode(i);
        else {
          var l = i - 65536;
          o += String.fromCharCode(55296 | l >> 10, 56320 | l & 1023);
        }
      }
      return o;
    }, Nt = (t, e) => t ? gn(U, t, e) : "";
    function wn(t, e) {
      e = I(e);
      var r = e === "std::string";
      Q(t, { name: e, fromWireType: function(n) {
        var a = F[n >> 2], o = n + 4, i;
        if (r)
          for (var s = o, c = 0; c <= a; ++c) {
            var l = o + c;
            if (c == a || U[l] == 0) {
              var d = l - s, v = Nt(s, d);
              i === void 0 ? i = v : (i += String.fromCharCode(0), i += v), s = l + 1;
            }
          }
        else {
          for (var m = new Array(a), c = 0; c < a; ++c)
            m[c] = String.fromCharCode(U[o + c]);
          i = m.join("");
        }
        return K(n), i;
      }, toWireType: function(n, a) {
        a instanceof ArrayBuffer && (a = new Uint8Array(a));
        var o, i = typeof a == "string";
        i || a instanceof Uint8Array || a instanceof Uint8ClampedArray || a instanceof Int8Array || _("Cannot pass non-string to std::string"), r && i ? o = Ce(a) : o = a.length;
        var s = Xt(4 + o + 1), c = s + 4;
        if (F[s >> 2] = o, r && i)
          mn(a, c, o + 1);
        else if (i)
          for (var l = 0; l < o; ++l) {
            var d = a.charCodeAt(l);
            d > 255 && (K(c), _("String has UTF-16 code units that do not fit in 8 bits")), U[c + l] = d;
          }
        else
          for (var l = 0; l < o; ++l)
            U[c + l] = a[l];
        return n !== null && n.push(K, s), s;
      }, argPackAdvance: 8, readValueFromPointer: dt, destructorFunction: function(n) {
        K(n);
      } });
    }
    var Te = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0, $n = (t, e) => {
      for (var r = t, n = r >> 1, a = n + e / 2; !(n >= a) && Ct[n]; )
        ++n;
      if (r = n << 1, r - t > 32 && Te)
        return Te.decode(U.subarray(t, r));
      for (var o = "", i = 0; !(i >= e / 2); ++i) {
        var s = ut[t + i * 2 >> 1];
        if (s == 0)
          break;
        o += String.fromCharCode(s);
      }
      return o;
    }, bn = (t, e, r) => {
      if (r === void 0 && (r = 2147483647), r < 2)
        return 0;
      r -= 2;
      for (var n = e, a = r < t.length * 2 ? r / 2 : t.length, o = 0; o < a; ++o) {
        var i = t.charCodeAt(o);
        ut[e >> 1] = i, e += 2;
      }
      return ut[e >> 1] = 0, e - n;
    }, Cn = (t) => t.length * 2, _n = (t, e) => {
      for (var r = 0, n = ""; !(r >= e / 4); ) {
        var a = R[t + r * 4 >> 2];
        if (a == 0)
          break;
        if (++r, a >= 65536) {
          var o = a - 65536;
          n += String.fromCharCode(55296 | o >> 10, 56320 | o & 1023);
        } else
          n += String.fromCharCode(a);
      }
      return n;
    }, Tn = (t, e, r) => {
      if (r === void 0 && (r = 2147483647), r < 4)
        return 0;
      for (var n = e, a = n + r - 4, o = 0; o < t.length; ++o) {
        var i = t.charCodeAt(o);
        if (i >= 55296 && i <= 57343) {
          var s = t.charCodeAt(++o);
          i = 65536 + ((i & 1023) << 10) | s & 1023;
        }
        if (R[e >> 2] = i, e += 4, e + 4 > a)
          break;
      }
      return R[e >> 2] = 0, e - n;
    }, Pn = (t) => {
      for (var e = 0, r = 0; r < t.length; ++r) {
        var n = t.charCodeAt(r);
        n >= 55296 && n <= 57343 && ++r, e += 4;
      }
      return e;
    }, An = function(t, e, r) {
      r = I(r);
      var n, a, o, i, s;
      e === 2 ? (n = $n, a = bn, i = Cn, o = () => Ct, s = 1) : e === 4 && (n = _n, a = Tn, i = Pn, o = () => F, s = 2), Q(t, { name: r, fromWireType: function(c) {
        for (var l = F[c >> 2], d = o(), v, m = c + 4, b = 0; b <= l; ++b) {
          var E = c + 4 + b * e;
          if (b == l || d[E >> s] == 0) {
            var W = E - m, x = n(m, W);
            v === void 0 ? v = x : (v += String.fromCharCode(0), v += x), m = E + e;
          }
        }
        return K(c), v;
      }, toWireType: function(c, l) {
        typeof l != "string" && _(`Cannot pass non-string to C++ string type ${r}`);
        var d = i(l), v = Xt(4 + d + e);
        return F[v >> 2] = d >> s, a(l, v + 4, d + e), c !== null && c.push(K, v), v;
      }, argPackAdvance: 8, readValueFromPointer: dt, destructorFunction: function(c) {
        K(c);
      } });
    };
    function En(t, e, r, n, a, o) {
      Et[t] = { name: I(e), rawConstructor: L(r, n), rawDestructor: L(a, o), fields: [] };
    }
    function xn(t, e, r, n, a, o, i, s, c, l) {
      Et[t].fields.push({ fieldName: I(e), getterReturnType: r, getter: L(n, a), getterContext: o, setterArgumentType: i, setter: L(s, c), setterContext: l });
    }
    function Sn(t, e) {
      e = I(e), Q(t, { isVoid: !0, name: e, argPackAdvance: 0, fromWireType: function() {
      }, toWireType: function(r, n) {
      } });
    }
    var Fn = {};
    function Mn(t) {
      var e = Fn[t];
      return e === void 0 ? I(t) : e;
    }
    function Pe() {
      if (typeof globalThis == "object")
        return globalThis;
      function t(e) {
        e.$$$embind_global$$$ = e;
        var r = typeof $$$embind_global$$$ == "object" && e.$$$embind_global$$$ == e;
        return r || delete e.$$$embind_global$$$, r;
      }
      if (typeof $$$embind_global$$$ == "object" || (typeof global == "object" && t(global) ? $$$embind_global$$$ = global : typeof self == "object" && t(self) && ($$$embind_global$$$ = self), typeof $$$embind_global$$$ == "object"))
        return $$$embind_global$$$;
      throw Error("unable to get global object.");
    }
    function Dn(t) {
      return t === 0 ? et.toHandle(Pe()) : (t = Mn(t), et.toHandle(Pe()[t]));
    }
    function On(t) {
      t > 4 && (B.get(t).refcount += 1);
    }
    function Ae(t, e) {
      var r = ot[t];
      return r === void 0 && _(e + " has unknown type " + ge(t)), r;
    }
    function jn(t) {
      var e = new Array(t + 1);
      return function(r, n, a) {
        e[0] = r;
        for (var o = 0; o < t; ++o) {
          var i = Ae(F[n + o * 4 >> 2], "parameter " + o);
          e[o + 1] = i.readValueFromPointer(a), a += i.argPackAdvance;
        }
        var s = new (r.bind.apply(r, e))();
        return et.toHandle(s);
      };
    }
    var Ee = {};
    function Rn(t, e, r, n) {
      t = et.toValue(t);
      var a = Ee[e];
      return a || (a = jn(e), Ee[e] = a), a(t, r, n);
    }
    function Wn(t, e) {
      t = Ae(t, "_emval_take_value");
      var r = t.readValueFromPointer(e);
      return et.toHandle(r);
    }
    var kn = () => {
      _t("");
    }, In = (t, e, r) => U.copyWithin(t, e, e + r), Hn = () => 2147483648, Un = (t) => {
      var e = X.buffer, r = t - e.byteLength + 65535 >>> 16;
      try {
        return X.grow(r), Qt(), 1;
      } catch {
      }
    }, zn = (t) => {
      var e = U.length;
      t >>>= 0;
      var r = Hn();
      if (t > r)
        return !1;
      for (var n = (c, l) => c + (l - c % l) % l, a = 1; a <= 4; a *= 2) {
        var o = e * (1 + 0.2 / a);
        o = Math.min(o, t + 100663296);
        var i = Math.min(r, n(Math.max(t, o), 65536)), s = Un(i);
        if (s)
          return !0;
      }
      return !1;
    }, Gt = {}, Vn = () => O || "./this.program", gt = () => {
      if (!gt.strings) {
        var t = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8", e = { USER: "web_user", LOGNAME: "web_user", PATH: "/", PWD: "/", HOME: "/home/web_user", LANG: t, _: Vn() };
        for (var r in Gt)
          Gt[r] === void 0 ? delete e[r] : e[r] = Gt[r];
        var n = [];
        for (var r in e)
          n.push(`${r}=${e[r]}`);
        gt.strings = n;
      }
      return gt.strings;
    }, Bn = (t, e) => {
      for (var r = 0; r < t.length; ++r)
        q[e++ >> 0] = t.charCodeAt(r);
      q[e >> 0] = 0;
    }, Yn = (t, e) => {
      var r = 0;
      return gt().forEach(function(n, a) {
        var o = e + r;
        F[t + a * 4 >> 2] = o, Bn(n, o), r += n.length + 1;
      }), 0;
    }, Ln = (t, e) => {
      var r = gt();
      F[t >> 2] = r.length;
      var n = 0;
      return r.forEach(function(a) {
        n += a.length + 1;
      }), F[e >> 2] = n, 0;
    };
    function Nn(t) {
      return t;
    }
    var Ot = (t) => t % 4 === 0 && (t % 100 !== 0 || t % 400 === 0), Gn = (t, e) => {
      for (var r = 0, n = 0; n <= e; r += t[n++])
        ;
      return r;
    }, xe = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Se = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Xn = (t, e) => {
      for (var r = new Date(t.getTime()); e > 0; ) {
        var n = Ot(r.getFullYear()), a = r.getMonth(), o = (n ? xe : Se)[a];
        if (e > o - r.getDate())
          e -= o - r.getDate() + 1, r.setDate(1), a < 11 ? r.setMonth(a + 1) : (r.setMonth(0), r.setFullYear(r.getFullYear() + 1));
        else
          return r.setDate(r.getDate() + e), r;
      }
      return r;
    };
    function qn(t, e, r) {
      var n = r > 0 ? r : Ce(t) + 1, a = new Array(n), o = be(t, a, 0, a.length);
      return e && (a.length = o), a;
    }
    var Jn = (t, e) => {
      q.set(t, e);
    }, Qn = (t, e, r, n) => {
      var a = R[n + 40 >> 2], o = { tm_sec: R[n >> 2], tm_min: R[n + 4 >> 2], tm_hour: R[n + 8 >> 2], tm_mday: R[n + 12 >> 2], tm_mon: R[n + 16 >> 2], tm_year: R[n + 20 >> 2], tm_wday: R[n + 24 >> 2], tm_yday: R[n + 28 >> 2], tm_isdst: R[n + 32 >> 2], tm_gmtoff: R[n + 36 >> 2], tm_zone: a ? Nt(a) : "" }, i = Nt(r), s = { "%c": "%a %b %d %H:%M:%S %Y", "%D": "%m/%d/%y", "%F": "%Y-%m-%d", "%h": "%b", "%r": "%I:%M:%S %p", "%R": "%H:%M", "%T": "%H:%M:%S", "%x": "%m/%d/%y", "%X": "%H:%M:%S", "%Ec": "%c", "%EC": "%C", "%Ex": "%m/%d/%y", "%EX": "%H:%M:%S", "%Ey": "%y", "%EY": "%Y", "%Od": "%d", "%Oe": "%e", "%OH": "%H", "%OI": "%I", "%Om": "%m", "%OM": "%M", "%OS": "%S", "%Ou": "%u", "%OU": "%U", "%OV": "%V", "%Ow": "%w", "%OW": "%W", "%Oy": "%y" };
      for (var c in s)
        i = i.replace(new RegExp(c, "g"), s[c]);
      var l = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"], d = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
      function v(h, y, S) {
        for (var D = typeof h == "number" ? h.toString() : h || ""; D.length < y; )
          D = S[0] + D;
        return D;
      }
      function m(h, y) {
        return v(h, y, "0");
      }
      function b(h, y) {
        function S(st) {
          return st < 0 ? -1 : st > 0 ? 1 : 0;
        }
        var D;
        return (D = S(h.getFullYear() - y.getFullYear())) === 0 && (D = S(h.getMonth() - y.getMonth())) === 0 && (D = S(h.getDate() - y.getDate())), D;
      }
      function E(h) {
        switch (h.getDay()) {
          case 0:
            return new Date(h.getFullYear() - 1, 11, 29);
          case 1:
            return h;
          case 2:
            return new Date(h.getFullYear(), 0, 3);
          case 3:
            return new Date(h.getFullYear(), 0, 2);
          case 4:
            return new Date(h.getFullYear(), 0, 1);
          case 5:
            return new Date(h.getFullYear() - 1, 11, 31);
          case 6:
            return new Date(h.getFullYear() - 1, 11, 30);
        }
      }
      function W(h) {
        var y = Xn(new Date(h.tm_year + 1900, 0, 1), h.tm_yday), S = new Date(y.getFullYear(), 0, 4), D = new Date(y.getFullYear() + 1, 0, 4), st = E(S), Rt = E(D);
        return b(st, y) <= 0 ? b(Rt, y) <= 0 ? y.getFullYear() + 1 : y.getFullYear() : y.getFullYear() - 1;
      }
      var x = { "%a": (h) => l[h.tm_wday].substring(0, 3), "%A": (h) => l[h.tm_wday], "%b": (h) => d[h.tm_mon].substring(0, 3), "%B": (h) => d[h.tm_mon], "%C": (h) => {
        var y = h.tm_year + 1900;
        return m(y / 100 | 0, 2);
      }, "%d": (h) => m(h.tm_mday, 2), "%e": (h) => v(h.tm_mday, 2, " "), "%g": (h) => W(h).toString().substring(2), "%G": (h) => W(h), "%H": (h) => m(h.tm_hour, 2), "%I": (h) => {
        var y = h.tm_hour;
        return y == 0 ? y = 12 : y > 12 && (y -= 12), m(y, 2);
      }, "%j": (h) => m(h.tm_mday + Gn(Ot(h.tm_year + 1900) ? xe : Se, h.tm_mon - 1), 3), "%m": (h) => m(h.tm_mon + 1, 2), "%M": (h) => m(h.tm_min, 2), "%n": () => `
`, "%p": (h) => h.tm_hour >= 0 && h.tm_hour < 12 ? "AM" : "PM", "%S": (h) => m(h.tm_sec, 2), "%t": () => "	", "%u": (h) => h.tm_wday || 7, "%U": (h) => {
        var y = h.tm_yday + 7 - h.tm_wday;
        return m(Math.floor(y / 7), 2);
      }, "%V": (h) => {
        var y = Math.floor((h.tm_yday + 7 - (h.tm_wday + 6) % 7) / 7);
        if ((h.tm_wday + 371 - h.tm_yday - 2) % 7 <= 2 && y++, y) {
          if (y == 53) {
            var S = (h.tm_wday + 371 - h.tm_yday) % 7;
            S != 4 && (S != 3 || !Ot(h.tm_year)) && (y = 1);
          }
        } else {
          y = 52;
          var D = (h.tm_wday + 7 - h.tm_yday - 1) % 7;
          (D == 4 || D == 5 && Ot(h.tm_year % 400 - 1)) && y++;
        }
        return m(y, 2);
      }, "%w": (h) => h.tm_wday, "%W": (h) => {
        var y = h.tm_yday + 7 - (h.tm_wday + 6) % 7;
        return m(Math.floor(y / 7), 2);
      }, "%y": (h) => (h.tm_year + 1900).toString().substring(2), "%Y": (h) => h.tm_year + 1900, "%z": (h) => {
        var y = h.tm_gmtoff, S = y >= 0;
        return y = Math.abs(y) / 60, y = y / 60 * 100 + y % 60, (S ? "+" : "-") + ("0000" + y).slice(-4);
      }, "%Z": (h) => h.tm_zone, "%%": () => "%" };
      i = i.replace(/%%/g, "\0\0");
      for (var c in x)
        i.includes(c) && (i = i.replace(new RegExp(c, "g"), x[c](o)));
      i = i.replace(/\0\0/g, "%");
      var k = qn(i, !1);
      return k.length > e ? 0 : (Jn(k, t), k.length - 1);
    }, Zn = (t, e, r, n, a) => Qn(t, e, r, n);
    ie = u.InternalError = class extends Error {
      constructor(t) {
        super(t), this.name = "InternalError";
      }
    }, Pr(), ft = u.BindingError = class extends Error {
      constructor(t) {
        super(t), this.name = "BindingError";
      }
    }, Vr(), jr(), Zr(), me = u.UnboundTypeError = rn(Error, "UnboundTypeError"), sn(), cn();
    var Kn = { q: dr, u: pr, a: vr, h: mr, l: gr, I: wr, P: $r, n: br, ba: Cr, d: yr, oa: _r, Y: Tr, fa: Er, na: nn, ma: an, D: on, ea: fn, W: hn, J: dn, w: yn, s: vn, V: wn, L: An, Q: En, pa: xn, ga: Sn, U: $e, la: Dn, R: On, ia: Rn, ka: Wn, K: kn, da: In, ca: zn, $: Yn, aa: Ln, H: ga, T: Ea, B: $a, p: ya, b: ta, C: ma, ha: Ca, c: oa, j: sa, i: na, x: wa, O: va, v: da, G: Ta, N: Pa, A: ba, F: xa, Z: Fa, X: Ma, k: ia, f: aa, e: ra, g: ea, M: Aa, m: ha, o: ua, S: ca, t: la, ja: pa, y: _a, r: fa, E: Sa, z: Nn, _: Zn };
    hr();
    var K = u._free = (t) => (K = u._free = A.sa)(t), Xt = u._malloc = (t) => (Xt = u._malloc = A.ta)(t), Fe = (t) => (Fe = A.va)(t);
    u.__embind_initialize_bindings = () => (u.__embind_initialize_bindings = A.wa)();
    var g = (t, e) => (g = A.xa)(t, e), wt = (t) => (wt = A.ya)(t), w = () => (w = A.za)(), $ = (t) => ($ = A.Aa)(t), Me = (t) => (Me = A.Ba)(t), De = (t) => (De = A.Ca)(t), Oe = (t, e, r) => (Oe = A.Da)(t, e, r), je = (t) => (je = A.Ea)(t);
    u.dynCall_viijii = (t, e, r, n, a, o, i) => (u.dynCall_viijii = A.Fa)(t, e, r, n, a, o, i);
    var Re = u.dynCall_jiii = (t, e, r, n) => (Re = u.dynCall_jiii = A.Ga)(t, e, r, n), We = u.dynCall_jiiii = (t, e, r, n, a) => (We = u.dynCall_jiiii = A.Ha)(t, e, r, n, a);
    u.dynCall_iiiiij = (t, e, r, n, a, o, i) => (u.dynCall_iiiiij = A.Ia)(t, e, r, n, a, o, i), u.dynCall_iiiiijj = (t, e, r, n, a, o, i, s, c) => (u.dynCall_iiiiijj = A.Ja)(t, e, r, n, a, o, i, s, c), u.dynCall_iiiiiijj = (t, e, r, n, a, o, i, s, c, l) => (u.dynCall_iiiiiijj = A.Ka)(t, e, r, n, a, o, i, s, c, l);
    function ta(t, e) {
      var r = w();
      try {
        return T(t)(e);
      } catch (n) {
        if ($(r), n !== n + 0)
          throw n;
        g(1, 0);
      }
    }
    function ea(t, e, r, n) {
      var a = w();
      try {
        T(t)(e, r, n);
      } catch (o) {
        if ($(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function ra(t, e, r) {
      var n = w();
      try {
        T(t)(e, r);
      } catch (a) {
        if ($(n), a !== a + 0)
          throw a;
        g(1, 0);
      }
    }
    function na(t, e, r, n, a) {
      var o = w();
      try {
        return T(t)(e, r, n, a);
      } catch (i) {
        if ($(o), i !== i + 0)
          throw i;
        g(1, 0);
      }
    }
    function aa(t, e) {
      var r = w();
      try {
        T(t)(e);
      } catch (n) {
        if ($(r), n !== n + 0)
          throw n;
        g(1, 0);
      }
    }
    function oa(t, e, r) {
      var n = w();
      try {
        return T(t)(e, r);
      } catch (a) {
        if ($(n), a !== a + 0)
          throw a;
        g(1, 0);
      }
    }
    function ia(t) {
      var e = w();
      try {
        T(t)();
      } catch (r) {
        if ($(e), r !== r + 0)
          throw r;
        g(1, 0);
      }
    }
    function sa(t, e, r, n) {
      var a = w();
      try {
        return T(t)(e, r, n);
      } catch (o) {
        if ($(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function ua(t, e, r, n, a, o) {
      var i = w();
      try {
        T(t)(e, r, n, a, o);
      } catch (s) {
        if ($(i), s !== s + 0)
          throw s;
        g(1, 0);
      }
    }
    function ca(t, e, r, n, a, o, i) {
      var s = w();
      try {
        T(t)(e, r, n, a, o, i);
      } catch (c) {
        if ($(s), c !== c + 0)
          throw c;
        g(1, 0);
      }
    }
    function fa(t, e, r, n, a, o, i, s, c, l, d) {
      var v = w();
      try {
        T(t)(e, r, n, a, o, i, s, c, l, d);
      } catch (m) {
        if ($(v), m !== m + 0)
          throw m;
        g(1, 0);
      }
    }
    function la(t, e, r, n, a, o, i, s) {
      var c = w();
      try {
        T(t)(e, r, n, a, o, i, s);
      } catch (l) {
        if ($(c), l !== l + 0)
          throw l;
        g(1, 0);
      }
    }
    function ha(t, e, r, n, a) {
      var o = w();
      try {
        T(t)(e, r, n, a);
      } catch (i) {
        if ($(o), i !== i + 0)
          throw i;
        g(1, 0);
      }
    }
    function da(t, e, r, n, a, o, i) {
      var s = w();
      try {
        return T(t)(e, r, n, a, o, i);
      } catch (c) {
        if ($(s), c !== c + 0)
          throw c;
        g(1, 0);
      }
    }
    function pa(t, e, r, n, a, o, i, s, c) {
      var l = w();
      try {
        T(t)(e, r, n, a, o, i, s, c);
      } catch (d) {
        if ($(l), d !== d + 0)
          throw d;
        g(1, 0);
      }
    }
    function ya(t) {
      var e = w();
      try {
        return T(t)();
      } catch (r) {
        if ($(e), r !== r + 0)
          throw r;
        g(1, 0);
      }
    }
    function va(t, e, r, n, a, o, i) {
      var s = w();
      try {
        return T(t)(e, r, n, a, o, i);
      } catch (c) {
        if ($(s), c !== c + 0)
          throw c;
        g(1, 0);
      }
    }
    function ma(t, e, r, n) {
      var a = w();
      try {
        return T(t)(e, r, n);
      } catch (o) {
        if ($(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function ga(t, e, r, n) {
      var a = w();
      try {
        return T(t)(e, r, n);
      } catch (o) {
        if ($(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function wa(t, e, r, n, a, o) {
      var i = w();
      try {
        return T(t)(e, r, n, a, o);
      } catch (s) {
        if ($(i), s !== s + 0)
          throw s;
        g(1, 0);
      }
    }
    function $a(t, e, r, n, a, o) {
      var i = w();
      try {
        return T(t)(e, r, n, a, o);
      } catch (s) {
        if ($(i), s !== s + 0)
          throw s;
        g(1, 0);
      }
    }
    function ba(t, e, r, n, a, o, i, s, c, l) {
      var d = w();
      try {
        return T(t)(e, r, n, a, o, i, s, c, l);
      } catch (v) {
        if ($(d), v !== v + 0)
          throw v;
        g(1, 0);
      }
    }
    function Ca(t, e, r) {
      var n = w();
      try {
        return T(t)(e, r);
      } catch (a) {
        if ($(n), a !== a + 0)
          throw a;
        g(1, 0);
      }
    }
    function _a(t, e, r, n, a, o, i, s, c, l) {
      var d = w();
      try {
        T(t)(e, r, n, a, o, i, s, c, l);
      } catch (v) {
        if ($(d), v !== v + 0)
          throw v;
        g(1, 0);
      }
    }
    function Ta(t, e, r, n, a, o, i, s) {
      var c = w();
      try {
        return T(t)(e, r, n, a, o, i, s);
      } catch (l) {
        if ($(c), l !== l + 0)
          throw l;
        g(1, 0);
      }
    }
    function Pa(t, e, r, n, a, o, i, s, c) {
      var l = w();
      try {
        return T(t)(e, r, n, a, o, i, s, c);
      } catch (d) {
        if ($(l), d !== d + 0)
          throw d;
        g(1, 0);
      }
    }
    function Aa(t, e, r, n, a, o, i) {
      var s = w();
      try {
        T(t)(e, r, n, a, o, i);
      } catch (c) {
        if ($(s), c !== c + 0)
          throw c;
        g(1, 0);
      }
    }
    function Ea(t, e, r, n) {
      var a = w();
      try {
        return T(t)(e, r, n);
      } catch (o) {
        if ($(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function xa(t, e, r, n, a, o, i, s, c, l, d, v) {
      var m = w();
      try {
        return T(t)(e, r, n, a, o, i, s, c, l, d, v);
      } catch (b) {
        if ($(m), b !== b + 0)
          throw b;
        g(1, 0);
      }
    }
    function Sa(t, e, r, n, a, o, i, s, c, l, d, v, m, b, E, W) {
      var x = w();
      try {
        T(t)(e, r, n, a, o, i, s, c, l, d, v, m, b, E, W);
      } catch (k) {
        if ($(x), k !== k + 0)
          throw k;
        g(1, 0);
      }
    }
    function Fa(t, e, r, n) {
      var a = w();
      try {
        return Re(t, e, r, n);
      } catch (o) {
        if ($(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function Ma(t, e, r, n, a) {
      var o = w();
      try {
        return We(t, e, r, n, a);
      } catch (i) {
        if ($(o), i !== i + 0)
          throw i;
        g(1, 0);
      }
    }
    var jt;
    ht = function t() {
      jt || ke(), jt || (ht = t);
    };
    function ke() {
      if (nt > 0 || (er(), nt > 0))
        return;
      function t() {
        jt || (jt = !0, u.calledRun = !0, !bt && (rr(), C(u), u.onRuntimeInitialized && u.onRuntimeInitialized(), nr()));
      }
      u.setStatus ? (u.setStatus("Running..."), setTimeout(function() {
        setTimeout(function() {
          u.setStatus("");
        }, 1), t();
      }, 1)) : t();
    }
    if (u.preInit)
      for (typeof u.preInit == "function" && (u.preInit = [u.preInit]); u.preInit.length > 0; )
        u.preInit.pop()();
    return ke(), p.ready;
  };
})();
function Ne(f) {
  return Ze(Ke, f);
}
async function qa(f, {
  tryHarder: p = rt.tryHarder,
  formats: u = rt.formats,
  maxSymbols: C = rt.maxSymbols
} = rt) {
  return Na(
    f,
    {
      tryHarder: p,
      formats: u,
      maxSymbols: C
    },
    Ke
  );
}
const tr = /* @__PURE__ */ new Map([
  ["aztec", "Aztec"],
  ["code_128", "Code128"],
  ["code_39", "Code39"],
  ["code_93", "Code93"],
  ["codabar", "Codabar"],
  ["data_matrix", "DataMatrix"],
  ["ean_13", "EAN-13"],
  ["ean_8", "EAN-8"],
  ["itf", "ITF"],
  ["pdf417", "PDF417"],
  ["qr_code", "QRCode"],
  ["upc_a", "UPC-A"],
  ["upc_e", "UPC-E"]
]);
function Ja(f) {
  for (const [p, u] of tr)
    if (f === u)
      return p;
  return "unknown";
}
var $t;
class Ka {
  constructor(p = {}) {
    Ue(this, $t, void 0);
    var u, C;
    if (((u = p == null ? void 0 : p.formats) == null ? void 0 : u.length) === 0)
      throw new TypeError("formats cannot be empty");
    (C = p == null ? void 0 : p.formats) == null || C.forEach((P) => {
      if (P === "unknown")
        throw new TypeError("format unknown is not supported");
      if (!Ve.includes(P))
        throw new TypeError(`format ${P} is not supported`);
    }), Ne(), ze(this, $t, (p == null ? void 0 : p.formats) ?? []);
  }
  static async getSupportedFormats() {
    return await Ne(), Ve.filter((p) => p !== "unknown");
  }
  async detect(p) {
    const u = await Ua(p);
    return u === null ? [] : (await qa(u, {
      tryHarder: !0,
      formats: He(this, $t).map(
        (P) => tr.get(P)
      )
    })).map((P) => {
      const {
        topLeft: { x: M, y: O },
        topRight: { x: N, y: H },
        bottomLeft: { x: j, y: Y },
        bottomRight: { x: G, y: z }
      } = P.position, V = Math.min(M, N, j, G), X = Math.min(O, H, Y, z), A = Math.max(M, N, j, G), bt = Math.max(O, H, Y, z);
      return {
        boundingBox: new DOMRectReadOnly(V, X, A - V, bt - X),
        rawValue: P.text,
        format: Ja(P.format),
        cornerPoints: [
          {
            x: M,
            y: O
          },
          {
            x: N,
            y: H
          },
          {
            x: G,
            y: z
          },
          {
            x: j,
            y: Y
          }
        ]
      };
    });
  }
}
$t = new WeakMap();
export {
  Ka as BarcodeDetector,
  Za as setZXingModuleOverrides
};
