// @sec-ant/zxing-wasm/dist/full/index.js
import { g as hi, r as _i, d as z, a as gi, w as yi, c as K } from "../core-5eb9bd55.js";
import { Z as Fi, b as Si, p as Ri, s as ki } from "../core-5eb9bd55.js";
var ke = (() => {
  var H = import.meta.url;
  return function(V = {}) {
    var f = V, B, Q;
    f.ready = new Promise((e, r) => {
      B = e, Q = r;
    });
    var de = Object.assign({}, f), pe = "./this.program", ze = typeof window == "object", he = typeof importScripts == "function";
    typeof process == "object" && typeof process.versions == "object" && process.versions.node;
    var W = "";
    function Ur(e) {
      return f.locateFile ? f.locateFile(e, W) : W + e;
    }
    var De;
    (ze || he) && (he ? W = self.location.href : typeof document < "u" && document.currentScript && (W = document.currentScript.src), H && (W = H), W.indexOf("blob:") !== 0 ? W = W.substr(0, W.replace(/[?#].*/, "").lastIndexOf("/") + 1) : W = "", he && (De = (e) => {
      var r = new XMLHttpRequest();
      return r.open("GET", e, !1), r.responseType = "arraybuffer", r.send(null), new Uint8Array(r.response);
    })), f.print || console.log.bind(console);
    var ne = f.printErr || console.error.bind(console);
    Object.assign(f, de), de = null, f.arguments && f.arguments, f.thisProgram && (pe = f.thisProgram), f.quit && f.quit;
    var ie;
    f.wasmBinary && (ie = f.wasmBinary), f.noExitRuntime, typeof WebAssembly != "object" && ye("no native wasm support detected");
    var _e, P, Xe = !1, U, D, ee, ge, F, A, Ze, Ge;
    function qe() {
      var e = _e.buffer;
      f.HEAP8 = U = new Int8Array(e), f.HEAP16 = ee = new Int16Array(e), f.HEAP32 = F = new Int32Array(e), f.HEAPU8 = D = new Uint8Array(e), f.HEAPU16 = ge = new Uint16Array(e), f.HEAPU32 = A = new Uint32Array(e), f.HEAPF32 = Ze = new Float32Array(e), f.HEAPF64 = Ge = new Float64Array(e);
    }
    var Je, Ke = [], Qe = [], er = [];
    function Mr() {
      if (f.preRun)
        for (typeof f.preRun == "function" && (f.preRun = [f.preRun]); f.preRun.length; )
          Yr(f.preRun.shift());
      Oe(Ke);
    }
    function Ir() {
      Oe(Qe);
    }
    function xr() {
      if (f.postRun)
        for (typeof f.postRun == "function" && (f.postRun = [f.postRun]); f.postRun.length; )
          Br(f.postRun.shift());
      Oe(er);
    }
    function Yr(e) {
      Ke.unshift(e);
    }
    function Vr(e) {
      Qe.unshift(e);
    }
    function Br(e) {
      er.unshift(e);
    }
    var X = 0, ae = null;
    function Lr(e) {
      X++, f.monitorRunDependencies && f.monitorRunDependencies(X);
    }
    function Nr(e) {
      if (X--, f.monitorRunDependencies && f.monitorRunDependencies(X), X == 0 && ae) {
        var r = ae;
        ae = null, r();
      }
    }
    function ye(e) {
      f.onAbort && f.onAbort(e), e = "Aborted(" + e + ")", ne(e), Xe = !0, e += ". Build with -sASSERTIONS for more info.";
      var r = new WebAssembly.RuntimeError(e);
      throw Q(r), r;
    }
    var zr = "data:application/octet-stream;base64,";
    function rr(e) {
      return e.startsWith(zr);
    }
    var Z;
    f.locateFile ? (Z = "zxing_full.wasm", rr(Z) || (Z = Ur(Z))) : Z = new URL("/full/zxing_full.wasm", self.location).href;
    function tr(e) {
      if (e == Z && ie)
        return new Uint8Array(ie);
      if (De)
        return De(e);
      throw "both async and sync fetching of the wasm failed";
    }
    function Xr(e) {
      return !ie && (ze || he) && typeof fetch == "function" ? fetch(e, { credentials: "same-origin" }).then((r) => {
        if (!r.ok)
          throw "failed to load wasm binary file at '" + e + "'";
        return r.arrayBuffer();
      }).catch(() => tr(e)) : Promise.resolve().then(() => tr(e));
    }
    function nr(e, r, t) {
      return Xr(e).then((n) => WebAssembly.instantiate(n, r)).then((n) => n).then(t, (n) => {
        ne("failed to asynchronously prepare wasm: " + n), ye(n);
      });
    }
    function Zr(e, r, t, n) {
      return !e && typeof WebAssembly.instantiateStreaming == "function" && !rr(r) && typeof fetch == "function" ? fetch(r, { credentials: "same-origin" }).then((i) => {
        var a = WebAssembly.instantiateStreaming(i, t);
        return a.then(n, function(o) {
          return ne("wasm streaming compile failed: " + o), ne("falling back to ArrayBuffer instantiation"), nr(r, t, n);
        });
      }) : nr(r, t, n);
    }
    function Gr() {
      var e = { a: Wn };
      function r(n, i) {
        var a = n.exports;
        return P = a, _e = P.qa, qe(), Je = P.ua, Vr(P.ra), Nr(), a;
      }
      Lr();
      function t(n) {
        r(n.instance);
      }
      if (f.instantiateWasm)
        try {
          return f.instantiateWasm(e, r);
        } catch (n) {
          ne("Module.instantiateWasm callback failed with error: " + n), Q(n);
        }
      return Zr(ie, Z, e, t).catch(Q), {};
    }
    var Oe = (e) => {
      for (; e.length > 0; )
        e.shift()(f);
    }, me = [], we = 0;
    function qr(e) {
      var r = new be(e);
      return r.get_caught() || (r.set_caught(!0), we--), r.set_rethrown(!1), me.push(r), kr(r.excPtr), r.get_exception_ptr();
    }
    var M = 0;
    function Jr() {
      _(0, 0);
      var e = me.pop();
      Rr(e.excPtr), M = 0;
    }
    function be(e) {
      this.excPtr = e, this.ptr = e - 24, this.set_type = function(r) {
        A[this.ptr + 4 >> 2] = r;
      }, this.get_type = function() {
        return A[this.ptr + 4 >> 2];
      }, this.set_destructor = function(r) {
        A[this.ptr + 8 >> 2] = r;
      }, this.get_destructor = function() {
        return A[this.ptr + 8 >> 2];
      }, this.set_caught = function(r) {
        r = r ? 1 : 0, U[this.ptr + 12 >> 0] = r;
      }, this.get_caught = function() {
        return U[this.ptr + 12 >> 0] != 0;
      }, this.set_rethrown = function(r) {
        r = r ? 1 : 0, U[this.ptr + 13 >> 0] = r;
      }, this.get_rethrown = function() {
        return U[this.ptr + 13 >> 0] != 0;
      }, this.init = function(r, t) {
        this.set_adjusted_ptr(0), this.set_type(r), this.set_destructor(t);
      }, this.set_adjusted_ptr = function(r) {
        A[this.ptr + 16 >> 2] = r;
      }, this.get_adjusted_ptr = function() {
        return A[this.ptr + 16 >> 2];
      }, this.get_exception_ptr = function() {
        var r = Or(this.get_type());
        if (r)
          return A[this.excPtr >> 2];
        var t = this.get_adjusted_ptr();
        return t !== 0 ? t : this.excPtr;
      };
    }
    function Kr(e) {
      throw M || (M = e), M;
    }
    var je = (e) => {
      var r = M;
      if (!r)
        return ve(0), 0;
      var t = new be(r);
      t.set_adjusted_ptr(r);
      var n = t.get_type();
      if (!n)
        return ve(0), r;
      for (var i in e) {
        var a = e[i];
        if (a === 0 || a === n)
          break;
        var o = t.ptr + 16;
        if (Dr(a, n, o))
          return ve(a), r;
      }
      return ve(n), r;
    }, Qr = () => je([]), et = (e) => je([e]), rt = (e, r) => je([e, r]);
    function tt(e) {
      var r = new be(e).get_exception_ptr();
      return r;
    }
    function nt() {
      var e = me.pop();
      e || ye("no exception to throw");
      var r = e.excPtr;
      throw e.get_rethrown() || (me.push(e), e.set_rethrown(!0), e.set_caught(!1), we++), M = r, M;
    }
    function it(e, r, t) {
      var n = new be(e);
      throw n.init(r, t), M = e, we++, M;
    }
    function at() {
      return we;
    }
    var $e = {};
    function ir(e) {
      for (; e.length; ) {
        var r = e.pop(), t = e.pop();
        t(r);
      }
    }
    function oe(e) {
      return this.fromWireType(F[e >> 2]);
    }
    var re = {}, G = {}, Ce = {}, ar = void 0;
    function Pe(e) {
      throw new ar(e);
    }
    function q(e, r, t) {
      e.forEach(function(s) {
        Ce[s] = r;
      });
      function n(s) {
        var u = t(s);
        u.length !== e.length && Pe("Mismatched type converter count");
        for (var c = 0; c < e.length; ++c)
          I(e[c], u[c]);
      }
      var i = new Array(r.length), a = [], o = 0;
      r.forEach((s, u) => {
        G.hasOwnProperty(s) ? i[u] = G[s] : (a.push(s), re.hasOwnProperty(s) || (re[s] = []), re[s].push(() => {
          i[u] = G[s], ++o, o === a.length && n(i);
        }));
      }), a.length === 0 && n(i);
    }
    var ot = function(e) {
      var r = $e[e];
      delete $e[e];
      var t = r.rawConstructor, n = r.rawDestructor, i = r.fields, a = i.map((o) => o.getterReturnType).concat(i.map((o) => o.setterArgumentType));
      q([e], a, (o) => {
        var s = {};
        return i.forEach((u, c) => {
          var v = u.fieldName, p = o[c], h = u.getter, m = u.getterContext, $ = o[c + i.length], S = u.setter, C = u.setterContext;
          s[v] = { read: (R) => p.fromWireType(h(m, R)), write: (R, l) => {
            var d = [];
            S(C, R, $.toWireType(d, l)), ir(d);
          } };
        }), [{ name: r.name, fromWireType: function(u) {
          var c = {};
          for (var v in s)
            c[v] = s[v].read(u);
          return n(u), c;
        }, toWireType: function(u, c) {
          for (var v in s)
            if (!(v in c))
              throw new TypeError(`Missing field: "${v}"`);
          var p = t();
          for (v in s)
            s[v].write(p, c[v]);
          return u !== null && u.push(n, p), p;
        }, argPackAdvance: 8, readValueFromPointer: oe, destructorFunction: n }];
      });
    };
    function st(e, r, t, n, i) {
    }
    function He(e) {
      switch (e) {
        case 1:
          return 0;
        case 2:
          return 1;
        case 4:
          return 2;
        case 8:
          return 3;
        default:
          throw new TypeError(`Unknown type size: ${e}`);
      }
    }
    function ut() {
      for (var e = new Array(256), r = 0; r < 256; ++r)
        e[r] = String.fromCharCode(r);
      or = e;
    }
    var or = void 0;
    function k(e) {
      for (var r = "", t = e; D[t]; )
        r += or[D[t++]];
      return r;
    }
    var te = void 0;
    function w(e) {
      throw new te(e);
    }
    function ft(e, r, t = {}) {
      var n = r.name;
      if (e || w(`type "${n}" must have a positive integer typeid pointer`), G.hasOwnProperty(e)) {
        if (t.ignoreDuplicateRegistrations)
          return;
        w(`Cannot register type '${n}' twice`);
      }
      if (G[e] = r, delete Ce[e], re.hasOwnProperty(e)) {
        var i = re[e];
        delete re[e], i.forEach((a) => a());
      }
    }
    function I(e, r, t = {}) {
      if (!("argPackAdvance" in r))
        throw new TypeError("registerType registeredInstance requires argPackAdvance");
      return ft(e, r, t);
    }
    function ct(e, r, t, n, i) {
      var a = He(t);
      r = k(r), I(e, { name: r, fromWireType: function(o) {
        return !!o;
      }, toWireType: function(o, s) {
        return s ? n : i;
      }, argPackAdvance: 8, readValueFromPointer: function(o) {
        var s;
        if (t === 1)
          s = U;
        else if (t === 2)
          s = ee;
        else if (t === 4)
          s = F;
        else
          throw new TypeError("Unknown boolean type size: " + r);
        return this.fromWireType(s[o >> a]);
      }, destructorFunction: null });
    }
    function lt(e) {
      if (!(this instanceof L) || !(e instanceof L))
        return !1;
      for (var r = this.$$.ptrType.registeredClass, t = this.$$.ptr, n = e.$$.ptrType.registeredClass, i = e.$$.ptr; r.baseClass; )
        t = r.upcast(t), r = r.baseClass;
      for (; n.baseClass; )
        i = n.upcast(i), n = n.baseClass;
      return r === n && t === i;
    }
    function vt(e) {
      return { count: e.count, deleteScheduled: e.deleteScheduled, preservePointerOnDelete: e.preservePointerOnDelete, ptr: e.ptr, ptrType: e.ptrType, smartPtr: e.smartPtr, smartPtrType: e.smartPtrType };
    }
    function We(e) {
      function r(t) {
        return t.$$.ptrType.registeredClass.name;
      }
      w(r(e) + " instance already deleted");
    }
    var Ue = !1;
    function sr(e) {
    }
    function dt(e) {
      e.smartPtr ? e.smartPtrType.rawDestructor(e.smartPtr) : e.ptrType.registeredClass.rawDestructor(e.ptr);
    }
    function ur(e) {
      e.count.value -= 1;
      var r = e.count.value === 0;
      r && dt(e);
    }
    function fr(e, r, t) {
      if (r === t)
        return e;
      if (t.baseClass === void 0)
        return null;
      var n = fr(e, r, t.baseClass);
      return n === null ? null : t.downcast(n);
    }
    var cr = {};
    function pt() {
      return Object.keys(fe).length;
    }
    function ht() {
      var e = [];
      for (var r in fe)
        fe.hasOwnProperty(r) && e.push(fe[r]);
      return e;
    }
    var se = [];
    function Me() {
      for (; se.length; ) {
        var e = se.pop();
        e.$$.deleteScheduled = !1, e.delete();
      }
    }
    var ue = void 0;
    function _t(e) {
      ue = e, se.length && ue && ue(Me);
    }
    function gt() {
      f.getInheritedInstanceCount = pt, f.getLiveInheritedInstances = ht, f.flushPendingDeletes = Me, f.setDelayFunction = _t;
    }
    var fe = {};
    function yt(e, r) {
      for (r === void 0 && w("ptr should not be undefined"); e.baseClass; )
        r = e.upcast(r), e = e.baseClass;
      return r;
    }
    function mt(e, r) {
      return r = yt(e, r), fe[r];
    }
    function Te(e, r) {
      (!r.ptrType || !r.ptr) && Pe("makeClassHandle requires ptr and ptrType");
      var t = !!r.smartPtrType, n = !!r.smartPtr;
      return t !== n && Pe("Both smartPtrType and smartPtr must be specified"), r.count = { value: 1 }, ce(Object.create(e, { $$: { value: r } }));
    }
    function wt(e) {
      var r = this.getPointee(e);
      if (!r)
        return this.destructor(e), null;
      var t = mt(this.registeredClass, r);
      if (t !== void 0) {
        if (t.$$.count.value === 0)
          return t.$$.ptr = r, t.$$.smartPtr = e, t.clone();
        var n = t.clone();
        return this.destructor(e), n;
      }
      function i() {
        return this.isSmartPointer ? Te(this.registeredClass.instancePrototype, { ptrType: this.pointeeType, ptr: r, smartPtrType: this, smartPtr: e }) : Te(this.registeredClass.instancePrototype, { ptrType: this, ptr: e });
      }
      var a = this.registeredClass.getActualType(r), o = cr[a];
      if (!o)
        return i.call(this);
      var s;
      this.isConst ? s = o.constPointerType : s = o.pointerType;
      var u = fr(r, this.registeredClass, s.registeredClass);
      return u === null ? i.call(this) : this.isSmartPointer ? Te(s.registeredClass.instancePrototype, { ptrType: s, ptr: u, smartPtrType: this, smartPtr: e }) : Te(s.registeredClass.instancePrototype, { ptrType: s, ptr: u });
    }
    var ce = function(e) {
      return typeof FinalizationRegistry > "u" ? (ce = (r) => r, e) : (Ue = new FinalizationRegistry((r) => {
        ur(r.$$);
      }), ce = (r) => {
        var t = r.$$, n = !!t.smartPtr;
        if (n) {
          var i = { $$: t };
          Ue.register(r, i, r);
        }
        return r;
      }, sr = (r) => Ue.unregister(r), ce(e));
    };
    function bt() {
      if (this.$$.ptr || We(this), this.$$.preservePointerOnDelete)
        return this.$$.count.value += 1, this;
      var e = ce(Object.create(Object.getPrototypeOf(this), { $$: { value: vt(this.$$) } }));
      return e.$$.count.value += 1, e.$$.deleteScheduled = !1, e;
    }
    function $t() {
      this.$$.ptr || We(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && w("Object already scheduled for deletion"), sr(this), ur(this.$$), this.$$.preservePointerOnDelete || (this.$$.smartPtr = void 0, this.$$.ptr = void 0);
    }
    function Ct() {
      return !this.$$.ptr;
    }
    function Pt() {
      return this.$$.ptr || We(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && w("Object already scheduled for deletion"), se.push(this), se.length === 1 && ue && ue(Me), this.$$.deleteScheduled = !0, this;
    }
    function Tt() {
      L.prototype.isAliasOf = lt, L.prototype.clone = bt, L.prototype.delete = $t, L.prototype.isDeleted = Ct, L.prototype.deleteLater = Pt;
    }
    function L() {
    }
    var At = 48, Et = 57;
    function lr(e) {
      if (e === void 0)
        return "_unknown";
      e = e.replace(/[^a-zA-Z0-9_]/g, "$");
      var r = e.charCodeAt(0);
      return r >= At && r <= Et ? `_${e}` : e;
    }
    function vr(e, r) {
      return e = lr(e), { [e]: function() {
        return r.apply(this, arguments);
      } }[e];
    }
    function dr(e, r, t) {
      if (e[r].overloadTable === void 0) {
        var n = e[r];
        e[r] = function() {
          return e[r].overloadTable.hasOwnProperty(arguments.length) || w(`Function '${t}' called with an invalid number of arguments (${arguments.length}) - expects one of (${e[r].overloadTable})!`), e[r].overloadTable[arguments.length].apply(this, arguments);
        }, e[r].overloadTable = [], e[r].overloadTable[n.argCount] = n;
      }
    }
    function pr(e, r, t) {
      f.hasOwnProperty(e) ? ((t === void 0 || f[e].overloadTable !== void 0 && f[e].overloadTable[t] !== void 0) && w(`Cannot register public name '${e}' twice`), dr(f, e, e), f.hasOwnProperty(t) && w(`Cannot register multiple overloads of a function with the same number of arguments (${t})!`), f[e].overloadTable[t] = r) : (f[e] = r, t !== void 0 && (f[e].numArguments = t));
    }
    function Ft(e, r, t, n, i, a, o, s) {
      this.name = e, this.constructor = r, this.instancePrototype = t, this.rawDestructor = n, this.baseClass = i, this.getActualType = a, this.upcast = o, this.downcast = s, this.pureVirtualFunctions = [];
    }
    function Ie(e, r, t) {
      for (; r !== t; )
        r.upcast || w(`Expected null or instance of ${t.name}, got an instance of ${r.name}`), e = r.upcast(e), r = r.baseClass;
      return e;
    }
    function St(e, r) {
      if (r === null)
        return this.isReference && w(`null is not a valid ${this.name}`), 0;
      r.$$ || w(`Cannot pass "${Ve(r)}" as a ${this.name}`), r.$$.ptr || w(`Cannot pass deleted object as a pointer of type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Ie(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function Rt(e, r) {
      var t;
      if (r === null)
        return this.isReference && w(`null is not a valid ${this.name}`), this.isSmartPointer ? (t = this.rawConstructor(), e !== null && e.push(this.rawDestructor, t), t) : 0;
      r.$$ || w(`Cannot pass "${Ve(r)}" as a ${this.name}`), r.$$.ptr || w(`Cannot pass deleted object as a pointer of type ${this.name}`), !this.isConst && r.$$.ptrType.isConst && w(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
      var n = r.$$.ptrType.registeredClass;
      if (t = Ie(r.$$.ptr, n, this.registeredClass), this.isSmartPointer)
        switch (r.$$.smartPtr === void 0 && w("Passing raw pointer to smart pointer is illegal"), this.sharingPolicy) {
          case 0:
            r.$$.smartPtrType === this ? t = r.$$.smartPtr : w(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
            break;
          case 1:
            t = r.$$.smartPtr;
            break;
          case 2:
            if (r.$$.smartPtrType === this)
              t = r.$$.smartPtr;
            else {
              var i = r.clone();
              t = this.rawShare(t, N.toHandle(function() {
                i.delete();
              })), e !== null && e.push(this.rawDestructor, t);
            }
            break;
          default:
            w("Unsupporting sharing policy");
        }
      return t;
    }
    function kt(e, r) {
      if (r === null)
        return this.isReference && w(`null is not a valid ${this.name}`), 0;
      r.$$ || w(`Cannot pass "${Ve(r)}" as a ${this.name}`), r.$$.ptr || w(`Cannot pass deleted object as a pointer of type ${this.name}`), r.$$.ptrType.isConst && w(`Cannot convert argument of type ${r.$$.ptrType.name} to parameter type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Ie(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function Dt(e) {
      return this.rawGetPointee && (e = this.rawGetPointee(e)), e;
    }
    function Ot(e) {
      this.rawDestructor && this.rawDestructor(e);
    }
    function jt(e) {
      e !== null && e.delete();
    }
    function Ht() {
      x.prototype.getPointee = Dt, x.prototype.destructor = Ot, x.prototype.argPackAdvance = 8, x.prototype.readValueFromPointer = oe, x.prototype.deleteObject = jt, x.prototype.fromWireType = wt;
    }
    function x(e, r, t, n, i, a, o, s, u, c, v) {
      this.name = e, this.registeredClass = r, this.isReference = t, this.isConst = n, this.isSmartPointer = i, this.pointeeType = a, this.sharingPolicy = o, this.rawGetPointee = s, this.rawConstructor = u, this.rawShare = c, this.rawDestructor = v, !i && r.baseClass === void 0 ? n ? (this.toWireType = St, this.destructorFunction = null) : (this.toWireType = kt, this.destructorFunction = null) : this.toWireType = Rt;
    }
    function hr(e, r, t) {
      f.hasOwnProperty(e) || Pe("Replacing nonexistant public symbol"), f[e].overloadTable !== void 0 && t !== void 0 ? f[e].overloadTable[t] = r : (f[e] = r, f[e].argCount = t);
    }
    var Wt = (e, r, t) => {
      var n = f["dynCall_" + e];
      return t && t.length ? n.apply(null, [r].concat(t)) : n.call(null, r);
    }, Ae = [], b = (e) => {
      var r = Ae[e];
      return r || (e >= Ae.length && (Ae.length = e + 1), Ae[e] = r = Je.get(e)), r;
    }, Ut = (e, r, t) => {
      if (e.includes("j"))
        return Wt(e, r, t);
      var n = b(r).apply(null, t);
      return n;
    }, Mt = (e, r) => {
      var t = [];
      return function() {
        return t.length = 0, Object.assign(t, arguments), Ut(e, r, t);
      };
    };
    function j(e, r) {
      e = k(e);
      function t() {
        return e.includes("j") ? Mt(e, r) : b(r);
      }
      var n = t();
      return typeof n != "function" && w(`unknown function pointer with signature ${e}: ${r}`), n;
    }
    function It(e, r) {
      var t = vr(r, function(n) {
        this.name = r, this.message = n;
        var i = new Error(n).stack;
        i !== void 0 && (this.stack = this.toString() + `
` + i.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return t.prototype = Object.create(e.prototype), t.prototype.constructor = t, t.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, t;
    }
    var _r = void 0;
    function gr(e) {
      var r = Sr(e), t = k(r);
      return Y(r), t;
    }
    function Ee(e, r) {
      var t = [], n = {};
      function i(a) {
        if (!n[a] && !G[a]) {
          if (Ce[a]) {
            Ce[a].forEach(i);
            return;
          }
          t.push(a), n[a] = !0;
        }
      }
      throw r.forEach(i), new _r(`${e}: ` + t.map(gr).join([", "]));
    }
    function xt(e, r, t, n, i, a, o, s, u, c, v, p, h) {
      v = k(v), a = j(i, a), s && (s = j(o, s)), c && (c = j(u, c)), h = j(p, h);
      var m = lr(v);
      pr(m, function() {
        Ee(`Cannot construct ${v} due to unbound types`, [n]);
      }), q([e, r, t], n ? [n] : [], function($) {
        $ = $[0];
        var S, C;
        n ? (S = $.registeredClass, C = S.instancePrototype) : C = L.prototype;
        var R = vr(m, function() {
          if (Object.getPrototypeOf(this) !== l)
            throw new te("Use 'new' to construct " + v);
          if (d.constructor_body === void 0)
            throw new te(v + " has no accessible constructor");
          var Re = d.constructor_body[arguments.length];
          if (Re === void 0)
            throw new te(`Tried to invoke ctor of ${v} with invalid number of parameters (${arguments.length}) - expected (${Object.keys(d.constructor_body).toString()}) parameters instead!`);
          return Re.apply(this, arguments);
        }), l = Object.create(C, { constructor: { value: R } });
        R.prototype = l;
        var d = new Ft(v, R, l, h, S, a, s, c);
        d.baseClass && (d.baseClass.__derivedClasses === void 0 && (d.baseClass.__derivedClasses = []), d.baseClass.__derivedClasses.push(d));
        var T = new x(v, d, !0, !1, !1), E = new x(v + "*", d, !1, !1, !1), J = new x(v + " const*", d, !1, !0, !1);
        return cr[e] = { pointerType: E, constPointerType: J }, hr(m, R), [T, E, J];
      });
    }
    function xe(e, r) {
      for (var t = [], n = 0; n < e; n++)
        t.push(A[r + n * 4 >> 2]);
      return t;
    }
    function Ye(e, r, t, n, i, a) {
      var o = r.length;
      o < 2 && w("argTypes array size mismatch! Must at least get return value and 'this' types!");
      for (var s = r[1] !== null && t !== null, u = !1, c = 1; c < r.length; ++c)
        if (r[c] !== null && r[c].destructorFunction === void 0) {
          u = !0;
          break;
        }
      var v = r[0].name !== "void", p = o - 2, h = new Array(p), m = [], $ = [];
      return function() {
        arguments.length !== p && w(`function ${e} called with ${arguments.length} arguments, expected ${p} args!`), $.length = 0;
        var S;
        m.length = s ? 2 : 1, m[0] = i, s && (S = r[1].toWireType($, this), m[1] = S);
        for (var C = 0; C < p; ++C)
          h[C] = r[C + 2].toWireType($, arguments[C]), m.push(h[C]);
        var R = n.apply(null, m);
        function l(d) {
          if (u)
            ir($);
          else
            for (var T = s ? 1 : 2; T < r.length; T++) {
              var E = T === 1 ? S : h[T - 2];
              r[T].destructorFunction !== null && r[T].destructorFunction(E);
            }
          if (v)
            return r[0].fromWireType(d);
        }
        return l(R);
      };
    }
    function Yt(e, r, t, n, i, a) {
      var o = xe(r, t);
      i = j(n, i), q([], [e], function(s) {
        s = s[0];
        var u = `constructor ${s.name}`;
        if (s.registeredClass.constructor_body === void 0 && (s.registeredClass.constructor_body = []), s.registeredClass.constructor_body[r - 1] !== void 0)
          throw new te(`Cannot register multiple constructors with identical number of parameters (${r - 1}) for class '${s.name}'! Overload resolution is currently only performed using the parameter count, not actual type info!`);
        return s.registeredClass.constructor_body[r - 1] = () => {
          Ee(`Cannot construct ${s.name} due to unbound types`, o);
        }, q([], o, function(c) {
          return c.splice(1, 0, null), s.registeredClass.constructor_body[r - 1] = Ye(u, c, null, i, a), [];
        }), [];
      });
    }
    function Vt(e, r, t, n, i, a, o, s, u) {
      var c = xe(t, n);
      r = k(r), a = j(i, a), q([], [e], function(v) {
        v = v[0];
        var p = `${v.name}.${r}`;
        r.startsWith("@@") && (r = Symbol[r.substring(2)]), s && v.registeredClass.pureVirtualFunctions.push(r);
        function h() {
          Ee(`Cannot call ${p} due to unbound types`, c);
        }
        var m = v.registeredClass.instancePrototype, $ = m[r];
        return $ === void 0 || $.overloadTable === void 0 && $.className !== v.name && $.argCount === t - 2 ? (h.argCount = t - 2, h.className = v.name, m[r] = h) : (dr(m, r, p), m[r].overloadTable[t - 2] = h), q([], c, function(S) {
          var C = Ye(p, S, v, a, o);
          return m[r].overloadTable === void 0 ? (C.argCount = t - 2, m[r] = C) : m[r].overloadTable[t - 2] = C, [];
        }), [];
      });
    }
    function Bt() {
      Object.assign(yr.prototype, { get(e) {
        return this.allocated[e];
      }, has(e) {
        return this.allocated[e] !== void 0;
      }, allocate(e) {
        var r = this.freelist.pop() || this.allocated.length;
        return this.allocated[r] = e, r;
      }, free(e) {
        this.allocated[e] = void 0, this.freelist.push(e);
      } });
    }
    function yr() {
      this.allocated = [void 0], this.freelist = [];
    }
    var O = new yr();
    function mr(e) {
      e >= O.reserved && --O.get(e).refcount === 0 && O.free(e);
    }
    function Lt() {
      for (var e = 0, r = O.reserved; r < O.allocated.length; ++r)
        O.allocated[r] !== void 0 && ++e;
      return e;
    }
    function Nt() {
      O.allocated.push({ value: void 0 }, { value: null }, { value: !0 }, { value: !1 }), O.reserved = O.allocated.length, f.count_emval_handles = Lt;
    }
    var N = { toValue: (e) => (e || w("Cannot use deleted val. handle = " + e), O.get(e).value), toHandle: (e) => {
      switch (e) {
        case void 0:
          return 1;
        case null:
          return 2;
        case !0:
          return 3;
        case !1:
          return 4;
        default:
          return O.allocate({ refcount: 1, value: e });
      }
    } };
    function zt(e, r) {
      r = k(r), I(e, { name: r, fromWireType: function(t) {
        var n = N.toValue(t);
        return mr(t), n;
      }, toWireType: function(t, n) {
        return N.toHandle(n);
      }, argPackAdvance: 8, readValueFromPointer: oe, destructorFunction: null });
    }
    function Ve(e) {
      if (e === null)
        return "null";
      var r = typeof e;
      return r === "object" || r === "array" || r === "function" ? e.toString() : "" + e;
    }
    function Xt(e, r) {
      switch (r) {
        case 2:
          return function(t) {
            return this.fromWireType(Ze[t >> 2]);
          };
        case 3:
          return function(t) {
            return this.fromWireType(Ge[t >> 3]);
          };
        default:
          throw new TypeError("Unknown float type: " + e);
      }
    }
    function Zt(e, r, t) {
      var n = He(t);
      r = k(r), I(e, { name: r, fromWireType: function(i) {
        return i;
      }, toWireType: function(i, a) {
        return a;
      }, argPackAdvance: 8, readValueFromPointer: Xt(r, n), destructorFunction: null });
    }
    function Gt(e, r, t, n, i, a, o) {
      var s = xe(r, t);
      e = k(e), i = j(n, i), pr(e, function() {
        Ee(`Cannot call ${e} due to unbound types`, s);
      }, r - 1), q([], s, function(u) {
        var c = [u[0], null].concat(u.slice(1));
        return hr(e, Ye(e, c, null, i, a), r - 1), [];
      });
    }
    function qt(e, r, t) {
      switch (r) {
        case 0:
          return t ? function(i) {
            return U[i];
          } : function(i) {
            return D[i];
          };
        case 1:
          return t ? function(i) {
            return ee[i >> 1];
          } : function(i) {
            return ge[i >> 1];
          };
        case 2:
          return t ? function(i) {
            return F[i >> 2];
          } : function(i) {
            return A[i >> 2];
          };
        default:
          throw new TypeError("Unknown integer type: " + e);
      }
    }
    function Jt(e, r, t, n, i) {
      r = k(r);
      var a = He(t), o = (p) => p;
      if (n === 0) {
        var s = 32 - 8 * t;
        o = (p) => p << s >>> s;
      }
      var u = r.includes("unsigned"), c = (p, h) => {
      }, v;
      u ? v = function(p, h) {
        return c(h, this.name), h >>> 0;
      } : v = function(p, h) {
        return c(h, this.name), h;
      }, I(e, { name: r, fromWireType: o, toWireType: v, argPackAdvance: 8, readValueFromPointer: qt(r, a, n !== 0), destructorFunction: null });
    }
    function Kt(e, r, t) {
      var n = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], i = n[r];
      function a(o) {
        o = o >> 2;
        var s = A, u = s[o], c = s[o + 1];
        return new i(s.buffer, c, u);
      }
      t = k(t), I(e, { name: t, fromWireType: a, argPackAdvance: 8, readValueFromPointer: a }, { ignoreDuplicateRegistrations: !0 });
    }
    var wr = (e, r, t, n) => {
      if (!(n > 0))
        return 0;
      for (var i = t, a = t + n - 1, o = 0; o < e.length; ++o) {
        var s = e.charCodeAt(o);
        if (s >= 55296 && s <= 57343) {
          var u = e.charCodeAt(++o);
          s = 65536 + ((s & 1023) << 10) | u & 1023;
        }
        if (s <= 127) {
          if (t >= a)
            break;
          r[t++] = s;
        } else if (s <= 2047) {
          if (t + 1 >= a)
            break;
          r[t++] = 192 | s >> 6, r[t++] = 128 | s & 63;
        } else if (s <= 65535) {
          if (t + 2 >= a)
            break;
          r[t++] = 224 | s >> 12, r[t++] = 128 | s >> 6 & 63, r[t++] = 128 | s & 63;
        } else {
          if (t + 3 >= a)
            break;
          r[t++] = 240 | s >> 18, r[t++] = 128 | s >> 12 & 63, r[t++] = 128 | s >> 6 & 63, r[t++] = 128 | s & 63;
        }
      }
      return r[t] = 0, t - i;
    }, Qt = (e, r, t) => wr(e, D, r, t), br = (e) => {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n <= 127 ? r++ : n <= 2047 ? r += 2 : n >= 55296 && n <= 57343 ? (r += 4, ++t) : r += 3;
      }
      return r;
    }, $r = typeof TextDecoder < "u" ? new TextDecoder("utf8") : void 0, en = (e, r, t) => {
      for (var n = r + t, i = r; e[i] && !(i >= n); )
        ++i;
      if (i - r > 16 && e.buffer && $r)
        return $r.decode(e.subarray(r, i));
      for (var a = ""; r < i; ) {
        var o = e[r++];
        if (!(o & 128)) {
          a += String.fromCharCode(o);
          continue;
        }
        var s = e[r++] & 63;
        if ((o & 224) == 192) {
          a += String.fromCharCode((o & 31) << 6 | s);
          continue;
        }
        var u = e[r++] & 63;
        if ((o & 240) == 224 ? o = (o & 15) << 12 | s << 6 | u : o = (o & 7) << 18 | s << 12 | u << 6 | e[r++] & 63, o < 65536)
          a += String.fromCharCode(o);
        else {
          var c = o - 65536;
          a += String.fromCharCode(55296 | c >> 10, 56320 | c & 1023);
        }
      }
      return a;
    }, Be = (e, r) => e ? en(D, e, r) : "";
    function rn(e, r) {
      r = k(r);
      var t = r === "std::string";
      I(e, { name: r, fromWireType: function(n) {
        var i = A[n >> 2], a = n + 4, o;
        if (t)
          for (var s = a, u = 0; u <= i; ++u) {
            var c = a + u;
            if (u == i || D[c] == 0) {
              var v = c - s, p = Be(s, v);
              o === void 0 ? o = p : (o += String.fromCharCode(0), o += p), s = c + 1;
            }
          }
        else {
          for (var h = new Array(i), u = 0; u < i; ++u)
            h[u] = String.fromCharCode(D[a + u]);
          o = h.join("");
        }
        return Y(n), o;
      }, toWireType: function(n, i) {
        i instanceof ArrayBuffer && (i = new Uint8Array(i));
        var a, o = typeof i == "string";
        o || i instanceof Uint8Array || i instanceof Uint8ClampedArray || i instanceof Int8Array || w("Cannot pass non-string to std::string"), t && o ? a = br(i) : a = i.length;
        var s = Ne(4 + a + 1), u = s + 4;
        if (A[s >> 2] = a, t && o)
          Qt(i, u, a + 1);
        else if (o)
          for (var c = 0; c < a; ++c) {
            var v = i.charCodeAt(c);
            v > 255 && (Y(u), w("String has UTF-16 code units that do not fit in 8 bits")), D[u + c] = v;
          }
        else
          for (var c = 0; c < a; ++c)
            D[u + c] = i[c];
        return n !== null && n.push(Y, s), s;
      }, argPackAdvance: 8, readValueFromPointer: oe, destructorFunction: function(n) {
        Y(n);
      } });
    }
    var Cr = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0, tn = (e, r) => {
      for (var t = e, n = t >> 1, i = n + r / 2; !(n >= i) && ge[n]; )
        ++n;
      if (t = n << 1, t - e > 32 && Cr)
        return Cr.decode(D.subarray(e, t));
      for (var a = "", o = 0; !(o >= r / 2); ++o) {
        var s = ee[e + o * 2 >> 1];
        if (s == 0)
          break;
        a += String.fromCharCode(s);
      }
      return a;
    }, nn = (e, r, t) => {
      if (t === void 0 && (t = 2147483647), t < 2)
        return 0;
      t -= 2;
      for (var n = r, i = t < e.length * 2 ? t / 2 : e.length, a = 0; a < i; ++a) {
        var o = e.charCodeAt(a);
        ee[r >> 1] = o, r += 2;
      }
      return ee[r >> 1] = 0, r - n;
    }, an = (e) => e.length * 2, on = (e, r) => {
      for (var t = 0, n = ""; !(t >= r / 4); ) {
        var i = F[e + t * 4 >> 2];
        if (i == 0)
          break;
        if (++t, i >= 65536) {
          var a = i - 65536;
          n += String.fromCharCode(55296 | a >> 10, 56320 | a & 1023);
        } else
          n += String.fromCharCode(i);
      }
      return n;
    }, sn = (e, r, t) => {
      if (t === void 0 && (t = 2147483647), t < 4)
        return 0;
      for (var n = r, i = n + t - 4, a = 0; a < e.length; ++a) {
        var o = e.charCodeAt(a);
        if (o >= 55296 && o <= 57343) {
          var s = e.charCodeAt(++a);
          o = 65536 + ((o & 1023) << 10) | s & 1023;
        }
        if (F[r >> 2] = o, r += 4, r + 4 > i)
          break;
      }
      return F[r >> 2] = 0, r - n;
    }, un = (e) => {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n >= 55296 && n <= 57343 && ++t, r += 4;
      }
      return r;
    }, fn = function(e, r, t) {
      t = k(t);
      var n, i, a, o, s;
      r === 2 ? (n = tn, i = nn, o = an, a = () => ge, s = 1) : r === 4 && (n = on, i = sn, o = un, a = () => A, s = 2), I(e, { name: t, fromWireType: function(u) {
        for (var c = A[u >> 2], v = a(), p, h = u + 4, m = 0; m <= c; ++m) {
          var $ = u + 4 + m * r;
          if (m == c || v[$ >> s] == 0) {
            var S = $ - h, C = n(h, S);
            p === void 0 ? p = C : (p += String.fromCharCode(0), p += C), h = $ + r;
          }
        }
        return Y(u), p;
      }, toWireType: function(u, c) {
        typeof c != "string" && w(`Cannot pass non-string to C++ string type ${t}`);
        var v = o(c), p = Ne(4 + v + r);
        return A[p >> 2] = v >> s, i(c, p + 4, v + r), u !== null && u.push(Y, p), p;
      }, argPackAdvance: 8, readValueFromPointer: oe, destructorFunction: function(u) {
        Y(u);
      } });
    };
    function cn(e, r, t, n, i, a) {
      $e[e] = { name: k(r), rawConstructor: j(t, n), rawDestructor: j(i, a), fields: [] };
    }
    function ln(e, r, t, n, i, a, o, s, u, c) {
      $e[e].fields.push({ fieldName: k(r), getterReturnType: t, getter: j(n, i), getterContext: a, setterArgumentType: o, setter: j(s, u), setterContext: c });
    }
    function vn(e, r) {
      r = k(r), I(e, { isVoid: !0, name: r, argPackAdvance: 0, fromWireType: function() {
      }, toWireType: function(t, n) {
      } });
    }
    var dn = {};
    function pn(e) {
      var r = dn[e];
      return r === void 0 ? k(e) : r;
    }
    function Pr() {
      if (typeof globalThis == "object")
        return globalThis;
      function e(r) {
        r.$$$embind_global$$$ = r;
        var t = typeof $$$embind_global$$$ == "object" && r.$$$embind_global$$$ == r;
        return t || delete r.$$$embind_global$$$, t;
      }
      if (typeof $$$embind_global$$$ == "object" || (typeof global == "object" && e(global) ? $$$embind_global$$$ = global : typeof self == "object" && e(self) && ($$$embind_global$$$ = self), typeof $$$embind_global$$$ == "object"))
        return $$$embind_global$$$;
      throw Error("unable to get global object.");
    }
    function hn(e) {
      return e === 0 ? N.toHandle(Pr()) : (e = pn(e), N.toHandle(Pr()[e]));
    }
    function _n(e) {
      e > 4 && (O.get(e).refcount += 1);
    }
    function Tr(e, r) {
      var t = G[e];
      return t === void 0 && w(r + " has unknown type " + gr(e)), t;
    }
    function gn(e) {
      var r = new Array(e + 1);
      return function(t, n, i) {
        r[0] = t;
        for (var a = 0; a < e; ++a) {
          var o = Tr(A[n + a * 4 >> 2], "parameter " + a);
          r[a + 1] = o.readValueFromPointer(i), i += o.argPackAdvance;
        }
        var s = new (t.bind.apply(t, r))();
        return N.toHandle(s);
      };
    }
    var Ar = {};
    function yn(e, r, t, n) {
      e = N.toValue(e);
      var i = Ar[r];
      return i || (i = gn(r), Ar[r] = i), i(e, t, n);
    }
    function mn(e, r) {
      e = Tr(e, "_emval_take_value");
      var t = e.readValueFromPointer(r);
      return N.toHandle(t);
    }
    var wn = () => {
      ye("");
    }, bn = (e, r, t) => D.copyWithin(e, r, r + t), $n = () => 2147483648, Cn = (e) => {
      var r = _e.buffer, t = e - r.byteLength + 65535 >>> 16;
      try {
        return _e.grow(t), qe(), 1;
      } catch {
      }
    }, Pn = (e) => {
      var r = D.length;
      e >>>= 0;
      var t = $n();
      if (e > t)
        return !1;
      for (var n = (u, c) => u + (c - u % c) % c, i = 1; i <= 4; i *= 2) {
        var a = r * (1 + 0.2 / i);
        a = Math.min(a, e + 100663296);
        var o = Math.min(t, n(Math.max(e, a), 65536)), s = Cn(o);
        if (s)
          return !0;
      }
      return !1;
    }, Le = {}, Tn = () => pe || "./this.program", le = () => {
      if (!le.strings) {
        var e = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8", r = { USER: "web_user", LOGNAME: "web_user", PATH: "/", PWD: "/", HOME: "/home/web_user", LANG: e, _: Tn() };
        for (var t in Le)
          Le[t] === void 0 ? delete r[t] : r[t] = Le[t];
        var n = [];
        for (var t in r)
          n.push(`${t}=${r[t]}`);
        le.strings = n;
      }
      return le.strings;
    }, An = (e, r) => {
      for (var t = 0; t < e.length; ++t)
        U[r++ >> 0] = e.charCodeAt(t);
      U[r >> 0] = 0;
    }, En = (e, r) => {
      var t = 0;
      return le().forEach(function(n, i) {
        var a = r + t;
        A[e + i * 4 >> 2] = a, An(n, a), t += n.length + 1;
      }), 0;
    }, Fn = (e, r) => {
      var t = le();
      A[e >> 2] = t.length;
      var n = 0;
      return t.forEach(function(i) {
        n += i.length + 1;
      }), A[r >> 2] = n, 0;
    };
    function Sn(e) {
      return e;
    }
    var Fe = (e) => e % 4 === 0 && (e % 100 !== 0 || e % 400 === 0), Rn = (e, r) => {
      for (var t = 0, n = 0; n <= r; t += e[n++])
        ;
      return t;
    }, Er = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Fr = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], kn = (e, r) => {
      for (var t = new Date(e.getTime()); r > 0; ) {
        var n = Fe(t.getFullYear()), i = t.getMonth(), a = (n ? Er : Fr)[i];
        if (r > a - t.getDate())
          r -= a - t.getDate() + 1, t.setDate(1), i < 11 ? t.setMonth(i + 1) : (t.setMonth(0), t.setFullYear(t.getFullYear() + 1));
        else
          return t.setDate(t.getDate() + r), t;
      }
      return t;
    };
    function Dn(e, r, t) {
      var n = t > 0 ? t : br(e) + 1, i = new Array(n), a = wr(e, i, 0, i.length);
      return r && (i.length = a), i;
    }
    var On = (e, r) => {
      U.set(e, r);
    }, jn = (e, r, t, n) => {
      var i = F[n + 40 >> 2], a = { tm_sec: F[n >> 2], tm_min: F[n + 4 >> 2], tm_hour: F[n + 8 >> 2], tm_mday: F[n + 12 >> 2], tm_mon: F[n + 16 >> 2], tm_year: F[n + 20 >> 2], tm_wday: F[n + 24 >> 2], tm_yday: F[n + 28 >> 2], tm_isdst: F[n + 32 >> 2], tm_gmtoff: F[n + 36 >> 2], tm_zone: i ? Be(i) : "" }, o = Be(t), s = { "%c": "%a %b %d %H:%M:%S %Y", "%D": "%m/%d/%y", "%F": "%Y-%m-%d", "%h": "%b", "%r": "%I:%M:%S %p", "%R": "%H:%M", "%T": "%H:%M:%S", "%x": "%m/%d/%y", "%X": "%H:%M:%S", "%Ec": "%c", "%EC": "%C", "%Ex": "%m/%d/%y", "%EX": "%H:%M:%S", "%Ey": "%y", "%EY": "%Y", "%Od": "%d", "%Oe": "%e", "%OH": "%H", "%OI": "%I", "%Om": "%m", "%OM": "%M", "%OS": "%S", "%Ou": "%u", "%OU": "%U", "%OV": "%V", "%Ow": "%w", "%OW": "%W", "%Oy": "%y" };
      for (var u in s)
        o = o.replace(new RegExp(u, "g"), s[u]);
      var c = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"], v = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
      function p(l, d, T) {
        for (var E = typeof l == "number" ? l.toString() : l || ""; E.length < d; )
          E = T[0] + E;
        return E;
      }
      function h(l, d) {
        return p(l, d, "0");
      }
      function m(l, d) {
        function T(J) {
          return J < 0 ? -1 : J > 0 ? 1 : 0;
        }
        var E;
        return (E = T(l.getFullYear() - d.getFullYear())) === 0 && (E = T(l.getMonth() - d.getMonth())) === 0 && (E = T(l.getDate() - d.getDate())), E;
      }
      function $(l) {
        switch (l.getDay()) {
          case 0:
            return new Date(l.getFullYear() - 1, 11, 29);
          case 1:
            return l;
          case 2:
            return new Date(l.getFullYear(), 0, 3);
          case 3:
            return new Date(l.getFullYear(), 0, 2);
          case 4:
            return new Date(l.getFullYear(), 0, 1);
          case 5:
            return new Date(l.getFullYear() - 1, 11, 31);
          case 6:
            return new Date(l.getFullYear() - 1, 11, 30);
        }
      }
      function S(l) {
        var d = kn(new Date(l.tm_year + 1900, 0, 1), l.tm_yday), T = new Date(d.getFullYear(), 0, 4), E = new Date(d.getFullYear() + 1, 0, 4), J = $(T), Re = $(E);
        return m(J, d) <= 0 ? m(Re, d) <= 0 ? d.getFullYear() + 1 : d.getFullYear() : d.getFullYear() - 1;
      }
      var C = { "%a": (l) => c[l.tm_wday].substring(0, 3), "%A": (l) => c[l.tm_wday], "%b": (l) => v[l.tm_mon].substring(0, 3), "%B": (l) => v[l.tm_mon], "%C": (l) => {
        var d = l.tm_year + 1900;
        return h(d / 100 | 0, 2);
      }, "%d": (l) => h(l.tm_mday, 2), "%e": (l) => p(l.tm_mday, 2, " "), "%g": (l) => S(l).toString().substring(2), "%G": (l) => S(l), "%H": (l) => h(l.tm_hour, 2), "%I": (l) => {
        var d = l.tm_hour;
        return d == 0 ? d = 12 : d > 12 && (d -= 12), h(d, 2);
      }, "%j": (l) => h(l.tm_mday + Rn(Fe(l.tm_year + 1900) ? Er : Fr, l.tm_mon - 1), 3), "%m": (l) => h(l.tm_mon + 1, 2), "%M": (l) => h(l.tm_min, 2), "%n": () => `
`, "%p": (l) => l.tm_hour >= 0 && l.tm_hour < 12 ? "AM" : "PM", "%S": (l) => h(l.tm_sec, 2), "%t": () => "	", "%u": (l) => l.tm_wday || 7, "%U": (l) => {
        var d = l.tm_yday + 7 - l.tm_wday;
        return h(Math.floor(d / 7), 2);
      }, "%V": (l) => {
        var d = Math.floor((l.tm_yday + 7 - (l.tm_wday + 6) % 7) / 7);
        if ((l.tm_wday + 371 - l.tm_yday - 2) % 7 <= 2 && d++, d) {
          if (d == 53) {
            var E = (l.tm_wday + 371 - l.tm_yday) % 7;
            E != 4 && (E != 3 || !Fe(l.tm_year)) && (d = 1);
          }
        } else {
          d = 52;
          var T = (l.tm_wday + 7 - l.tm_yday - 1) % 7;
          (T == 4 || T == 5 && Fe(l.tm_year % 400 - 1)) && d++;
        }
        return h(d, 2);
      }, "%w": (l) => l.tm_wday, "%W": (l) => {
        var d = l.tm_yday + 7 - (l.tm_wday + 6) % 7;
        return h(Math.floor(d / 7), 2);
      }, "%y": (l) => (l.tm_year + 1900).toString().substring(2), "%Y": (l) => l.tm_year + 1900, "%z": (l) => {
        var d = l.tm_gmtoff, T = d >= 0;
        return d = Math.abs(d) / 60, d = d / 60 * 100 + d % 60, (T ? "+" : "-") + ("0000" + d).slice(-4);
      }, "%Z": (l) => l.tm_zone, "%%": () => "%" };
      o = o.replace(/%%/g, "\0\0");
      for (var u in C)
        o.includes(u) && (o = o.replace(new RegExp(u, "g"), C[u](a)));
      o = o.replace(/\0\0/g, "%");
      var R = Dn(o, !1);
      return R.length > r ? 0 : (On(R, e), R.length - 1);
    }, Hn = (e, r, t, n, i) => jn(e, r, t, n);
    ar = f.InternalError = class extends Error {
      constructor(r) {
        super(r), this.name = "InternalError";
      }
    }, ut(), te = f.BindingError = class extends Error {
      constructor(r) {
        super(r), this.name = "BindingError";
      }
    }, Tt(), gt(), Ht(), _r = f.UnboundTypeError = It(Error, "UnboundTypeError"), Bt(), Nt();
    var Wn = { r: qr, v: Jr, a: Qr, h: et, m: rt, K: tt, R: nt, p: it, ca: at, d: Kr, oa: ot, Z: st, ga: ct, na: xt, ma: Yt, E: Vt, fa: zt, X: Zt, F: Gt, y: Jt, t: Kt, W: rn, N: fn, L: cn, pa: ln, ha: vn, V: mr, la: hn, S: _n, ja: yn, ka: mn, M: wn, ea: bn, da: Pn, aa: En, ba: Fn, J: ri, U: ci, C: ni, q: Kn, b: Un, D: ei, ia: ai, c: Vn, i: Ln, j: xn, o: ti, Q: Qn, w: Jn, I: si, P: ui, B: ii, H: li, _: di, Y: pi, k: Bn, e: Yn, f: In, g: Mn, O: fi, l: Gn, n: Nn, T: zn, u: Zn, x: qn, A: oi, s: Xn, G: vi, z: Sn, $: Hn };
    Gr();
    var Y = f._free = (e) => (Y = f._free = P.sa)(e), Ne = f._malloc = (e) => (Ne = f._malloc = P.ta)(e), Sr = (e) => (Sr = P.va)(e);
    f.__embind_initialize_bindings = () => (f.__embind_initialize_bindings = P.wa)();
    var _ = (e, r) => (_ = P.xa)(e, r), ve = (e) => (ve = P.ya)(e), g = () => (g = P.za)(), y = (e) => (y = P.Aa)(e), Rr = (e) => (Rr = P.Ba)(e), kr = (e) => (kr = P.Ca)(e), Dr = (e, r, t) => (Dr = P.Da)(e, r, t), Or = (e) => (Or = P.Ea)(e);
    f.dynCall_viijii = (e, r, t, n, i, a, o) => (f.dynCall_viijii = P.Fa)(e, r, t, n, i, a, o);
    var jr = f.dynCall_jiii = (e, r, t, n) => (jr = f.dynCall_jiii = P.Ga)(e, r, t, n), Hr = f.dynCall_jiiii = (e, r, t, n, i) => (Hr = f.dynCall_jiiii = P.Ha)(e, r, t, n, i);
    f.dynCall_iiiiij = (e, r, t, n, i, a, o) => (f.dynCall_iiiiij = P.Ia)(e, r, t, n, i, a, o), f.dynCall_iiiiijj = (e, r, t, n, i, a, o, s, u) => (f.dynCall_iiiiijj = P.Ja)(e, r, t, n, i, a, o, s, u), f.dynCall_iiiiiijj = (e, r, t, n, i, a, o, s, u, c) => (f.dynCall_iiiiiijj = P.Ka)(e, r, t, n, i, a, o, s, u, c);
    function Un(e, r) {
      var t = g();
      try {
        return b(e)(r);
      } catch (n) {
        if (y(t), n !== n + 0)
          throw n;
        _(1, 0);
      }
    }
    function Mn(e, r, t, n) {
      var i = g();
      try {
        b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function In(e, r, t) {
      var n = g();
      try {
        b(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function xn(e, r, t, n, i) {
      var a = g();
      try {
        return b(e)(r, t, n, i);
      } catch (o) {
        if (y(a), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function Yn(e, r) {
      var t = g();
      try {
        b(e)(r);
      } catch (n) {
        if (y(t), n !== n + 0)
          throw n;
        _(1, 0);
      }
    }
    function Vn(e, r, t) {
      var n = g();
      try {
        return b(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function Bn(e) {
      var r = g();
      try {
        b(e)();
      } catch (t) {
        if (y(r), t !== t + 0)
          throw t;
        _(1, 0);
      }
    }
    function Ln(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function Nn(e, r, t, n, i, a) {
      var o = g();
      try {
        b(e)(r, t, n, i, a);
      } catch (s) {
        if (y(o), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function zn(e, r, t, n, i, a, o) {
      var s = g();
      try {
        b(e)(r, t, n, i, a, o);
      } catch (u) {
        if (y(s), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function Xn(e, r, t, n, i, a, o, s, u, c, v) {
      var p = g();
      try {
        b(e)(r, t, n, i, a, o, s, u, c, v);
      } catch (h) {
        if (y(p), h !== h + 0)
          throw h;
        _(1, 0);
      }
    }
    function Zn(e, r, t, n, i, a, o, s) {
      var u = g();
      try {
        b(e)(r, t, n, i, a, o, s);
      } catch (c) {
        if (y(u), c !== c + 0)
          throw c;
        _(1, 0);
      }
    }
    function Gn(e, r, t, n, i) {
      var a = g();
      try {
        b(e)(r, t, n, i);
      } catch (o) {
        if (y(a), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function qn(e, r, t, n, i, a, o, s, u) {
      var c = g();
      try {
        b(e)(r, t, n, i, a, o, s, u);
      } catch (v) {
        if (y(c), v !== v + 0)
          throw v;
        _(1, 0);
      }
    }
    function Jn(e, r, t, n, i, a, o) {
      var s = g();
      try {
        return b(e)(r, t, n, i, a, o);
      } catch (u) {
        if (y(s), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function Kn(e) {
      var r = g();
      try {
        return b(e)();
      } catch (t) {
        if (y(r), t !== t + 0)
          throw t;
        _(1, 0);
      }
    }
    function Qn(e, r, t, n, i, a, o) {
      var s = g();
      try {
        return b(e)(r, t, n, i, a, o);
      } catch (u) {
        if (y(s), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function ei(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function ri(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function ti(e, r, t, n, i, a) {
      var o = g();
      try {
        return b(e)(r, t, n, i, a);
      } catch (s) {
        if (y(o), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function ni(e, r, t, n, i, a) {
      var o = g();
      try {
        return b(e)(r, t, n, i, a);
      } catch (s) {
        if (y(o), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function ii(e, r, t, n, i, a, o, s, u, c) {
      var v = g();
      try {
        return b(e)(r, t, n, i, a, o, s, u, c);
      } catch (p) {
        if (y(v), p !== p + 0)
          throw p;
        _(1, 0);
      }
    }
    function ai(e, r, t) {
      var n = g();
      try {
        return b(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function oi(e, r, t, n, i, a, o, s, u, c) {
      var v = g();
      try {
        b(e)(r, t, n, i, a, o, s, u, c);
      } catch (p) {
        if (y(v), p !== p + 0)
          throw p;
        _(1, 0);
      }
    }
    function si(e, r, t, n, i, a, o, s) {
      var u = g();
      try {
        return b(e)(r, t, n, i, a, o, s);
      } catch (c) {
        if (y(u), c !== c + 0)
          throw c;
        _(1, 0);
      }
    }
    function ui(e, r, t, n, i, a, o, s, u) {
      var c = g();
      try {
        return b(e)(r, t, n, i, a, o, s, u);
      } catch (v) {
        if (y(c), v !== v + 0)
          throw v;
        _(1, 0);
      }
    }
    function fi(e, r, t, n, i, a, o) {
      var s = g();
      try {
        b(e)(r, t, n, i, a, o);
      } catch (u) {
        if (y(s), u !== u + 0)
          throw u;
        _(1, 0);
      }
    }
    function ci(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function li(e, r, t, n, i, a, o, s, u, c, v, p) {
      var h = g();
      try {
        return b(e)(r, t, n, i, a, o, s, u, c, v, p);
      } catch (m) {
        if (y(h), m !== m + 0)
          throw m;
        _(1, 0);
      }
    }
    function vi(e, r, t, n, i, a, o, s, u, c, v, p, h, m, $, S) {
      var C = g();
      try {
        b(e)(r, t, n, i, a, o, s, u, c, v, p, h, m, $, S);
      } catch (R) {
        if (y(C), R !== R + 0)
          throw R;
        _(1, 0);
      }
    }
    function di(e, r, t, n) {
      var i = g();
      try {
        return jr(e, r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function pi(e, r, t, n, i) {
      var a = g();
      try {
        return Hr(e, r, t, n, i);
      } catch (o) {
        if (y(a), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    var Se;
    ae = function e() {
      Se || Wr(), Se || (ae = e);
    };
    function Wr() {
      if (X > 0 || (Mr(), X > 0))
        return;
      function e() {
        Se || (Se = !0, f.calledRun = !0, !Xe && (Ir(), B(f), f.onRuntimeInitialized && f.onRuntimeInitialized(), xr()));
      }
      f.setStatus ? (f.setStatus("Running..."), setTimeout(function() {
        setTimeout(function() {
          f.setStatus("");
        }, 1), e();
      }, 1)) : e();
    }
    if (f.preInit)
      for (typeof f.preInit == "function" && (f.preInit = [f.preInit]); f.preInit.length > 0; )
        f.preInit.pop()();
    return Wr(), V.ready;
  };
})();
function $i(H) {
  return hi(ke, H);
}
async function Ci(H, {
  tryHarder: V = z.tryHarder,
  formats: f = z.formats,
  maxSymbols: B = z.maxSymbols
} = z) {
  return _i(
    H,
    {
      tryHarder: V,
      formats: f,
      maxSymbols: B
    },
    ke
  );
}
async function Pi(H, {
  tryHarder: V = z.tryHarder,
  formats: f = z.formats,
  maxSymbols: B = z.maxSymbols
} = z) {
  return gi(
    H,
    {
      tryHarder: V,
      formats: f,
      maxSymbols: B
    },
    ke
  );
}
async function Ti(H, {
  format: V = K.format,
  charset: f = K.charset,
  quietZone: B = K.quietZone,
  width: Q = K.width,
  height: de = K.height,
  eccLevel: pe = K.eccLevel
} = K) {
  return yi(
    H,
    {
      format: V,
      charset: f,
      quietZone: B,
      width: Q,
      height: de,
      eccLevel: pe
    },
    ke
  );
}
export {
  Fi as ZXING_BARCODE_FORMAT_NAMES,
  Si as ZXING_CHARACTOR_SET_NAMES,
  z as defaultZXingReadOptions,
  K as defaultZXingWriteOptions,
  $i as getZXingModule,
  Ri as purgeZXingModule,
  Pi as readBarcodeFromImageData,
  Ci as readBarcodesFromImageFile,
  ki as setZXingModuleOverrides,
  Ti as writeBarcodeToImageFile
};
